<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | The MetaRbolomics book</title>
  <meta name="description" content="References | The MetaRbolomics book" />
  <meta name="generator" content="bookdown 0.13.2 and GitBook 2.6.7" />

  <meta property="og:title" content="References | The MetaRbolomics book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | The MetaRbolomics book" />
  
  
  

<meta name="author" content="Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="0300-conclusion.html"/>
<link rel="next" href="9900-Appendices.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.8/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-rowgroup-1.10.16/css/rowGroup.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-rowgroup-1.10.16/js/dataTables.rowGroup.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="libs/vis-4.20.1/vis.min.js"></script>
<script src="libs/visNetwork-binding-2.0.8/visNetwork.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The MetaRbolomics book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to"><i class="fa fa-check"></i>How to</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="0100-introduction.html"><a href="0100-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="0101-introduction-Metabolomics_data_processing_and_analysis.html"><a href="0101-introduction-Metabolomics_data_processing_and_analysis.html"><i class="fa fa-check"></i><b>1.1</b> Metabolomics data processing and analysis</a></li>
<li class="chapter" data-level="1.2" data-path="0102-introduction-The_R_package_landscape.html"><a href="0102-introduction-The_R_package_landscape.html"><i class="fa fa-check"></i><b>1.2</b> The R package landscape</a></li>
<li class="chapter" data-level="1.3" data-path="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><a href="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><i class="fa fa-check"></i><b>1.3</b> Dependences and connectivity of metabolomics packages</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="0200-R_packages_for_metabolomics.html"><a href="0200-R_packages_for_metabolomics.html"><i class="fa fa-check"></i><b>2</b> R-packages for metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.1</b> Mass spectrometry data handling and (pre-)processing</a><ul>
<li class="chapter" data-level="2.1.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#profile-mode-and-centroided-data"><i class="fa fa-check"></i><b>2.1.1</b> Profile mode and centroided data</a></li>
<li class="chapter" data-level="2.1.2" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#direct-infusion-mass-spectrometry-data"><i class="fa fa-check"></i><b>2.1.2</b> Direct infusion mass spectrometry data</a></li>
<li class="chapter" data-level="2.1.3" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#hyphenated-ms-and-non-targeted-data"><i class="fa fa-check"></i><b>2.1.3</b> Hyphenated MS and non-targeted data</a></li>
<li class="chapter" data-level="2.1.4" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#targeted-data-and-alternative-representations-of-data"><i class="fa fa-check"></i><b>2.1.4</b> Targeted data and alternative representations of data</a></li>
<li class="chapter" data-level="2.1.5" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#additional-dimensionality"><i class="fa fa-check"></i><b>2.1.5</b> Additional dimensionality</a></li>
<li class="chapter" data-level="2.1.6" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#structuring-data-and-metadata"><i class="fa fa-check"></i><b>2.1.6</b> Structuring data and metadata</a></li>
<li class="chapter" data-level="2.1.7" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#ion-species-grouping-and-annotation"><i class="fa fa-check"></i><b>2.1.7</b> Ion species grouping and annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><i class="fa fa-check"></i><b>2.2</b> Metabolite identification with MS/MS data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#msms-data-handling-spectral-matching-and-clustering"><i class="fa fa-check"></i><b>2.2.1</b> MS/MS data handling, spectral matching and clustering</a></li>
<li class="chapter" data-level="2.2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#reading-of-spectral-databases"><i class="fa fa-check"></i><b>2.2.2</b> Reading of spectral databases</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><a href="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.3</b> NMR data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.4" data-path="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><a href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.4</b> UV data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.5" data-path="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><a href="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><i class="fa fa-check"></i><b>2.5</b> Statistical analysis of metabolomics data</a></li>
<li class="chapter" data-level="2.6" data-path="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><a href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><i class="fa fa-check"></i><b>2.6</b> Handling of molecule structures and chemical structure databases</a></li>
<li class="chapter" data-level="2.7" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><i class="fa fa-check"></i><b>2.7</b> Network analysis and biochemical pathways</a><ul>
<li class="chapter" data-level="2.7.1" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#network-infrastructure-and-analysis"><i class="fa fa-check"></i><b>2.7.1</b> Network infrastructure and analysis</a></li>
<li class="chapter" data-level="2.7.2" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#metabolite-annotation"><i class="fa fa-check"></i><b>2.7.2</b> Metabolite annotation</a></li>
<li class="chapter" data-level="2.7.3" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#generation-of-metabolic-networks"><i class="fa fa-check"></i><b>2.7.3</b> Generation of metabolic networks</a></li>
<li class="chapter" data-level="2.7.4" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-analysis"><i class="fa fa-check"></i><b>2.7.4</b> Pathway analysis</a></li>
<li class="chapter" data-level="2.7.5" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-resources-and-interfaces"><i class="fa fa-check"></i><b>2.7.5</b> Pathway resources and interfaces</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><a href="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><i class="fa fa-check"></i><b>2.8</b> Multifunctional workflows</a></li>
<li class="chapter" data-level="2.9" data-path="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><a href="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><i class="fa fa-check"></i><b>2.9</b> User interfaces and workflow management systems</a></li>
<li class="chapter" data-level="2.10" data-path="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><a href="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><i class="fa fa-check"></i><b>2.10</b> Metabolomics data sets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="0300-conclusion.html"><a href="0300-conclusion.html"><i class="fa fa-check"></i><b>3</b> Conclusions</a></li>
<li class="chapter" data-level="" data-path="0400-references.html"><a href="0400-references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="4" data-path="9900-Appendices.html"><a href="9900-Appendices.html"><i class="fa fa-check"></i><b>4</b> Appendices</a><ul>
<li class="chapter" data-level="" data-path="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><a href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><i class="fa fa-check"></i>Appendix 1: The MSP File Format and package support</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><i class="fa fa-check"></i>Appendix 2: metaRbolomics dependencies network</a><ul>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#libraries-and-settings"><i class="fa fa-check"></i>Libraries and settings</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#read-package-names-from-our-table"><i class="fa fa-check"></i>Read package names from our table</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#get-reverse-dependencies"><i class="fa fa-check"></i>Get reverse dependencies</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#building-dependency-network"><i class="fa fa-check"></i>Building dependency network</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#save-network-plot"><i class="fa fa-check"></i>Save network plot</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#notes"><i class="fa fa-check"></i>Notes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The MetaRbolomics book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div style="page-break-before: always;" />


</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs">
<div id="ref-emwas_2019">
<p>1. Emwas, A.-H.; Roy, R.; McKay, R.T.; Tenori, L.; Saccenti, E.; Gowda, G.A.N.; Raftery, D.; Alahmari, F.; Jaremko, L.; Jaremko, M. et al. NMR spectroscopy for metabolomics research. <em>Metabolites</em> <strong>2019</strong>, <em>9</em>.</p>
</div>
<div id="ref-lammerhofer_2013">
<p>2. <em>Metabolomics in practice: Successful strategies to generate and analyze metabolic data</em>; Lämmerhofer, M., Weckwerth, W., Eds.; Wiley-VCH Verlag GmbH &amp; Co. KGaA: Weinheim, Germany, 2013; ISBN <a href="https://worldcat.org/isbn/9783527330898">9783527330898</a>.</p>
</div>
<div id="ref-villasboas_2007">
<p>3. Villas-Boas, S.G.; Nielsen, J.; Smedsgaard, J.; Hansen, M.A.E.; Roessner-Tunali, U. <em>Metabolome analysis: An introduction</em>; 1st ed.; Wiley, John &amp; Sons, Incorporated, 2007; p. 319; ISBN <a href="https://worldcat.org/isbn/978-0-471-74344-6">978-0-471-74344-6</a>.</p>
</div>
<div id="ref-wehrens_2019">
<p>4. <em>Metabolomics: Practical guide to design and analysis</em>; Wehrens, R., Salek, R., Eds.; Chapman &amp; hall/CRC mathematical and computational biology; Chapman; Hall/CRC, 2019; ISBN <a href="https://worldcat.org/isbn/1498725260">1498725260</a>.</p>
</div>
<div id="ref-InternationalMetabolomicsSociety_2019">
<p>5. International Metabolomics Society Free Tools &amp; Learning Resources - Metabolomics Society Wiki.</p>
</div>
<div id="ref-salek_website_nd">
<p>6. Salek, R.; Emery, L.; Beisken, S. Metabolomics: An introduction EMBL-EBI train online.</p>
</div>
<div id="ref-rcoredevelopmentteam_website_2018">
<p>7. R Core Development Team R: A language and environment for statistical computing 2018.</p>
</div>
<div id="ref-spicer_website_2018">
<p>8. Spicer, R. GitHub - RASpicer/MetabolomicsTools 2018.</p>
</div>
<div id="ref-spicer_2017">
<p>9. Spicer, R.; Salek, R.M.; Moreno, P.; Cañueto, D.; Steinbeck, C. Navigating freely-available software tools for metabolomics analysis. <em>Metabolomics : Official journal of the Metabolomic Society</em> <strong>2017</strong>, <em>13</em>, 106.</p>
</div>
<div id="ref-misra_2016">
<p>10. Misra, B.B.; Hooft, J.J.J. van der Updates in metabolomics tools and resources: 2014-2015. <em>Electrophoresis</em> <strong>2016</strong>, <em>37</em>, 86–110.</p>
</div>
<div id="ref-misra_2017">
<p>11. Misra, B.B.; Fahrmann, J.F.; Grapov, D. Review of emerging metabolomic tools and resources: 2015-2016. <em>Electrophoresis</em> <strong>2017</strong>, <em>38</em>, 2257–2274.</p>
</div>
<div id="ref-misra_2018">
<p>12. Misra, B.B. New tools and resources in metabolomics: 2016-2017. <em>Electrophoresis</em> <strong>2018</strong>, <em>39</em>, 909–923.</p>
</div>
<div id="ref-misra_website_2017">
<p>13. Misra, B. GitHub - biswapriyamisra/metabolomics: Tools databases resources in metabolomics &amp; integrated omics in 2015-2016 2017.</p>
</div>
<div id="ref-kannan_2016">
<p>14. Kannan, L.; Ramos, M.; Re, A.; El-Hachem, N.; Safikhani, Z.; Gendoo, D.M.A.; Davis, S.; Gomez-Cabrero, D.; Castelo, R.; Hansen, K.D. et al. Public data and open source tools for multi-assay genomic investigation of disease. <em>Briefings in Bioinformatics</em> <strong>2016</strong>, <em>17</em>, 603–615.</p>
</div>
<div id="ref-blaenovi_2018">
<p>15. Blaženović, I.; Kind, T.; Ji, J.; Fiehn, O. Software tools and approaches for compound identification of LC-MS/MS data in metabolomics. <em>Metabolites</em> <strong>2018</strong>, <em>8</em>.</p>
</div>
<div id="ref-mullen_website_2019">
<p>16. Mullen, K. CRAN task view: Chemometrics and computational physics 2019.</p>
</div>
<div id="ref-gentleman_2004">
<p>17. Gentleman, R.C.; Carey, V.J.; Bates, D.M.; Bolstad, B.; Dettling, M.; Dudoit, S.; Ellis, B.; Gautier, L.; Ge, Y.; Gentry, J. et al. Bioconductor: Open software development for computational biology and bioinformatics. <em>Genome Biology</em> <strong>2004</strong>, <em>5</em>, R80.</p>
</div>
<div id="ref-bioconductor_website_nda">
<p>18. Bioconductor Bioconductor - BiocViews.</p>
</div>
<div id="ref-thecomprehensiverarchivenetwork_website_nd">
<p>19. The Comprehensive R Archive Network CRAN repository policy.</p>
</div>
<div id="ref-bioconductor_website_nd">
<p>20. Bioconductor Bioconductor - developers.</p>
</div>
<div id="ref-theul_2009">
<p>21. Theußl, S.; Zeileis, A. Collaborative software development using r-forge. <em>The R journal</em> <strong>2009</strong>, <em>1</em>, 9.</p>
</div>
<div id="ref-boettiger_2015">
<p>22. Boettiger, C.; Chamberlain, S.; Hart, E.; Ram, K. Building software, building community: Lessons from the rOpenSci project. <em>Journal of open research software</em> <strong>2015</strong>, <em>3</em>.</p>
</div>
<div id="ref-devries_website_2015b">
<p>23. Vries, A. de; Rickert, J. The network structure of r packages on CRAN &amp; BioConductor 2015.</p>
</div>
<div id="ref-devries_website_2015a">
<p>24. Vries, A. de Differences in the network structure of CRAN and BioConductor (revolutions) 2015.</p>
</div>
<div id="ref-devries_website_2015">
<p>25. Vries, A. de GitHub - andrie/cran-network-structure: Scripts used for my UseR!2015 presentation on the network structure of CRAN 2015.</p>
</div>
<div id="ref-neumann_website_2019">
<p>26. Neumann, S. GitHub - sneumann/metaRbolomics: Metabolomics in r and bioconductor 2019.</p>
</div>
<div id="ref-chambers_2012">
<p>27. Chambers, M.C.; Maclean, B.; Burke, R.; Amodei, D.; Ruderman, D.L.; Neumann, S.; Gatto, L.; Fischer, B.; Pratt, B.; Egertson, J. et al. A cross-platform toolkit for mass spectrometry and proteomics. <em>Nature Biotechnology</em> <strong>2012</strong>, <em>30</em>, 918–920.</p>
</div>
<div id="ref-kessner_2008">
<p>28. Kessner, D.; Chambers, M.; Burke, R.; Agus, D.; Mallick, P. ProteoWizard: Open source software for rapid proteomics tools development. <em>Bioinformatics</em> <strong>2008</strong>, <em>24</em>, 2534–2536.</p>
</div>
<div id="ref-fuhrer_2011">
<p>29. Fuhrer, T.; Heer, D.; Begemann, B.; Zamboni, N. High-throughput, accurate mass metabolome profiling of cellular extracts by flow injection-time-of-flight mass spectrometry. <em>Analytical Chemistry</em> <strong>2011</strong>, <em>83</em>, 7074–7080.</p>
</div>
<div id="ref-mahieu_2016a">
<p>30. Mahieu, N.G.; Genenbacher, J.L.; Patti, G.J. A roadmap for the XCMS family of software solutions in metabolomics. <em>Current Opinion in Chemical Biology</em> <strong>2016</strong>, <em>30</em>, 87–93.</p>
</div>
<div id="ref-smith_2006">
<p>31. Smith, C.A.; Want, E.J.; O’Maille, G.; Abagyan, R.; Siuzdak, G. XCMS: Processing mass spectrometry data for metabolite profiling using nonlinear peak alignment, matching, and identification. <em>Analytical Chemistry</em> <strong>2006</strong>, <em>78</em>, 779–787.</p>
</div>
<div id="ref-tautenhahn_2008">
<p>32. Tautenhahn, R.; Böttcher, C.; Neumann, S. Highly sensitive feature detection for high resolution LC/MS. <em>BMC Bioinformatics</em> <strong>2008</strong>, <em>9</em>, 504.</p>
</div>
<div id="ref-conley_2014">
<p>33. Conley, C.J.; Smith, R.; Torgrip, R.J.O.; Taylor, R.M.; Tautenhahn, R.; Prince, J.T. Massifquant: Open-source kalman filter-based XC-MS isotope trace feature detection. <em>Bioinformatics</em> <strong>2014</strong>, <em>30</em>, 2636–2643.</p>
</div>
<div id="ref-huber_2015">
<p>34. Huber, W.; Carey, V.J.; Gentleman, R.; Anders, S.; Carlson, M.; Carvalho, B.S.; Bravo, H.C.; Davis, S.; Gatto, L.; Girke, T. et al. Orchestrating high-throughput genomic analysis with bioconductor. <em>Nature Methods</em> <strong>2015</strong>, <em>12</em>, 115–121.</p>
</div>
<div id="ref-martinmorgan_2017">
<p>35. Martin Morgan, V.O. SummarizedExperiment. <em>Bioconductor</em> <strong>2017</strong>.</p>
</div>
<div id="ref-zhu_website_2019">
<p>36. Zhu, C. Zhuchcn/metabase: A r package to store, manipulate, analyze, and visualize metabolomics data 2019.</p>
</div>
<div id="ref-hoffmann_2019">
<p>37. Hoffmann, N.; Rein, J.; Sachsenberg, T.; Hartler, J.; Haug, K.; Mayer, G.; Alka, O.; Dayalan, S.; Pearce, J.T.M.; Rocca-Serra, P. et al. mzTab-m: A data standard for sharing quantitative results in mass spectrometry metabolomics. <em>Analytical Chemistry</em> <strong>2019</strong>, <em>91</em>, 3302–3310.</p>
</div>
<div id="ref-scheltema_2011">
<p>38. Scheltema, R.A.; Jankevics, A.; Jansen, R.C.; Swertz, M.A.; Breitling, R. PeakML/mzMatch: A file format, java library, r library, and tool-chain for mass spectrometry data analysis. <em>Analytical Chemistry</em> <strong>2011</strong>, <em>83</em>, 2786–2793.</p>
</div>
<div id="ref-shahaf_2016">
<p>39. Shahaf, N.; Rogachev, I.; Heinig, U.; Meir, S.; Malitsky, S.; Battat, M.; Wyner, H.; Zheng, S.; Wehrens, R.; Aharoni, A. The WEIZMA spectral library for high-confidence metabolite identification. <em>Nature Communications</em> <strong>2016</strong>, <em>7</em>, 12423.</p>
</div>
<div id="ref-witting_website_nd">
<p>40. Witting, M. GitHub - michaelwitting/ms2dbworkflow.</p>
</div>
<div id="ref-horai_2010">
<p>41. Horai, H.; Arita, M.; Kanaya, S.; Nihei, Y.; Ikeda, T.; Suwa, K.; Ojima, Y.; Tanaka, K.; Tanaka, S.; Aoshima, K. et al. MassBank: A public repository for sharing mass spectral data for life sciences. <em>Journal of Mass Spectrometry</em> <strong>2010</strong>, <em>45</em>, 703–714.</p>
</div>
<div id="ref-mylonas_2009">
<p>42. Mylonas, R.; Mauron, Y.; Masselot, A.; Binz, P.-A.; Budin, N.; Fathi, M.; Viette, V.; Hochstrasser, D.F.; Lisacek, F. X-rank: A robust algorithm for small molecule identification using tandem mass spectrometry. <em>Analytical Chemistry</em> <strong>2009</strong>, <em>81</em>, 7604–7610.</p>
</div>
<div id="ref-collins_2016">
<p>43. Collins, J.R.; Edwards, B.R.; Fredricks, H.F.; Van Mooy, B.A.S. LOBSTAHS: An adduct-based lipidomics strategy for discovery and identification of oxidative stress biomarkers. <em>Analytical Chemistry</em> <strong>2016</strong>, <em>88</em>, 7154–7162.</p>
</div>
<div id="ref-koelmel_2017">
<p>44. Koelmel, J.P.; Kroeger, N.M.; Ulmer, C.Z.; Bowden, J.A.; Patterson, R.E.; Cochran, J.A.; Beecher, C.W.W.; Garrett, T.J.; Yost, R.A. LipidMatch: An automated workflow for rule-based lipid identification using untargeted high-resolution tandem mass spectrometry data. <em>BMC Bioinformatics</em> <strong>2017</strong>, <em>18</em>, 331.</p>
</div>
<div id="ref-alcorizabalaguer_2018">
<p>45. Alcoriza-Balaguer, M.I.; García-Cañaveras, J.C.; Lopez, A.; Conde, I.; Juan, O.; Carretero, J.; Lahoz, A. LipidMS: An r package for lipid annotation in untargeted liquid chromatography-data independent acquisition-mass spectrometry lipidomics. <em>Analytical Chemistry</em> <strong>2018</strong>, <em>91</em>, 836–845.</p>
</div>
<div id="ref-thenationalinstituteofstandardsandtechnology_website_2012">
<p>46. Standards, T.N.I. of; Technology Library conversion tool 2012.</p>
</div>
<div id="ref-tsugawa_2015">
<p>47. Tsugawa, H.; Cajka, T.; Kind, T.; Ma, Y.; Higgins, B.; Ikeda, K.; Kanazawa, M.; VanderGheynst, J.; Fiehn, O.; Arita, M. MS-DIAL: Data-independent MS/MS deconvolution for comprehensive metabolome analysis. <em>Nature Methods</em> <strong>2015</strong>, <em>12</em>, 523–526.</p>
</div>
<div id="ref-massbankofnorthamerica_website_nd">
<p>48. MassBank of North America MoNA downloads.</p>
</div>
<div id="ref-wang_2016">
<p>49. Wang, M.; Carver, J.J.; Phelan, V.V.; Sanchez, L.M.; Garg, N.; Peng, Y.; Nguyen, D.D.; Watrous, J.; Kapono, C.A.; Luzzatto-Knaan, T. et al. Sharing and community curation of mass spectrometry data with global natural products social molecular networking. <em>Nature Biotechnology</em> <strong>2016</strong>, <em>34</em>, 828–837.</p>
</div>
<div id="ref-jacob_2017">
<p>50. Jacob, D.; Deborde, C.; Lefebvre, M.; Maucourt, M.; Moing, A. NMRProcFlow: A graphical and interactive tool dedicated to 1D spectra processing for NMR-based metabolomics. <em>Metabolomics : Official journal of the Metabolomic Society</em> <strong>2017</strong>, <em>13</em>, 36.</p>
</div>
<div id="ref-dieterle_2006">
<p>51. Dieterle, F.; Ross, A.; Schlotterbeck, G.; Senn, H. Probabilistic quotient normalization as robust method to account for dilution of complex biological mixtures. Application in 1H NMR metabonomics. <em>Analytical Chemistry</em> <strong>2006</strong>, <em>78</em>, 4281–4290.</p>
</div>
<div id="ref-wishart_2009">
<p>52. Wishart, D.S.; Knox, C.; Guo, A.C.; Eisner, R.; Young, N.; Gautam, B.; Hau, D.D.; Psychogios, N.; Dong, E.; Bouatra, S. et al. HMDB: A knowledgebase for the human metabolome. <em>Nucleic Acids Research</em> <strong>2009</strong>, <em>37</em>, D603–10.</p>
</div>
<div id="ref-wishart_2013">
<p>53. Wishart, D.S.; Jewison, T.; Guo, A.C.; Wilson, M.; Knox, C.; Liu, Y.; Djoumbou, Y.; Mandal, R.; Aziat, F.; Dong, E. et al. HMDB 3.0–the human metabolome database in 2013. <em>Nucleic Acids Research</em> <strong>2013</strong>, <em>41</em>, D801–7.</p>
</div>
<div id="ref-wishart_2007">
<p>54. Wishart, D.S.; Tzur, D.; Knox, C.; Eisner, R.; Guo, A.C.; Young, N.; Cheng, D.; Jewell, K.; Arndt, D.; Sawhney, S. et al. HMDB: The human metabolome database. <em>Nucleic Acids Research</em> <strong>2007</strong>, <em>35</em>, D521–6.</p>
</div>
<div id="ref-wishart_2018">
<p>55. Wishart, D.S.; Feunang, Y.D.; Marcu, A.; Guo, A.C.; Liang, K.; Vázquez-Fresno, R.; Sajed, T.; Johnson, D.; Li, C.; Karu, N. et al. HMDB 4.0: The human metabolome database for 2018. <em>Nucleic Acids Research</em> <strong>2018</strong>, <em>46</em>, D608–D617.</p>
</div>
<div id="ref-beckonert_2007">
<p>56. Beckonert, O.; Keun, H.C.; Ebbels, T.M.D.; Bundy, J.; Holmes, E.; Lindon, J.C.; Nicholson, J.K. Metabolic profiling, metabolomic and metabonomic procedures for NMR spectroscopy of urine, plasma, serum and tissue extracts. <em>Nature Protocols</em> <strong>2007</strong>, <em>2</em>, 2692–2703.</p>
</div>
<div id="ref-pudakalakatti_2014">
<p>57. Pudakalakatti, S.M.; Dubey, A.; Jaipuria, G.; Shubhashree, U.; Adiga, S.K.; Moskau, D.; Atreya, H.S. A fast NMR method for resonance assignments: Application to metabolomics. <em>Journal of Biomolecular NMR</em> <strong>2014</strong>, <em>58</em>, 165–173.</p>
</div>
<div id="ref-ludwig_2010">
<p>58. Ludwig, C.; Viant, M.R. Two-dimensional j-resolved NMR spectroscopy: Review of a key methodology in the metabolomics toolbox. <em>Phytochemical Analysis</em> <strong>2010</strong>, <em>21</em>, 22–32.</p>
</div>
<div id="ref-gmez_2014">
<p>59. Gómez, J.; Brezmes, J.; Mallol, R.; Rodríguez, M.A.; Vinaixa, M.; Salek, R.M.; Correig, X.; Cañellas, N. Dolphin: A tool for automatic targeted metabolite profiling using 1D and 2D (1)H-NMR data. <em>Analytical and Bioanalytical Chemistry</em> <strong>2014</strong>, <em>406</em>, 7967–7976.</p>
</div>
<div id="ref-shinzawa_2012">
<p>60. Shinzawa, H.; Nishida, M.; Kanematsu, W.; Tanaka, T.; Suzuki, K.; Noda, I. Parallel factor (PARAFAC) kernel analysis of temperature- and composition-dependent NMR spectra of poly(lactic acid) nanocomposites. <em>The Analyst</em> <strong>2012</strong>, <em>137</em>, 1913–1921.</p>
</div>
<div id="ref-chen_2018">
<p>61. Chen, K.; Park, J.; Li, F.; Patil, S.M.; Keire, D.A. Chemometric methods to quantify 1D and 2D NMR spectral differences among similar protein therapeutics. <em>AAPS PharmSciTech</em> <strong>2018</strong>, <em>19</em>, 1011–1019.</p>
</div>
<div id="ref-pedersen_2006">
<p>62. Pedersen, H.T.; Dyrby, M.; Engelsen, S.; Bro, R. Application of multi-way analysis to 2D NMR data. In; Annual reports on NMR spectroscopy; Elsevier, 2006; Vol. 59, pp. 207–233 ISBN <a href="https://worldcat.org/isbn/9780125054591">9780125054591</a>.</p>
</div>
<div id="ref-hao_2012">
<p>63. Hao, J.; Astle, W.; De Iorio, M.; Ebbels, T.M.D. BATMAN–an r package for the automated quantification of metabolites from nuclear magnetic resonance spectra using a bayesian model. <em>Bioinformatics</em> <strong>2012</strong>, <em>28</em>, 2088–2090.</p>
</div>
<div id="ref-bioconductor_website_ndb">
<p>64. Bioconductor Bioconductor - BiocViews: packages found under StatisticalMethod.</p>
</div>
<div id="ref-groemping_website_2019">
<p>65. Groemping, U. CRAN task view: Design of experiments (DoE) &amp; analysis of experimental data 2019.</p>
</div>
<div id="ref-leisch_website_2019">
<p>66. Leisch, F.; Gruen, B. CRAN task view: Cluster analysis &amp; finite mixture models 2019.</p>
</div>
<div id="ref-hewson_website_2018">
<p>67. Hewson, P. CRAN task view: Multivariate statistics 2018.</p>
</div>
<div id="ref-hothorn_website_2019">
<p>68. Hothorn, T. CRAN task view: Machine learning &amp; statistical learning 2019.</p>
</div>
<div id="ref-thecomprehensiverarchivenetwork_website_nda">
<p>69. The Comprehensive R Archive Network CRAN task views.</p>
</div>
<div id="ref-bishop_2006">
<p>70. Bishop, C.M. <em>Pattern recognition and machine learning (information science and statistics)</em>; Springer: New York, 2006; p. 738; ISBN <a href="https://worldcat.org/isbn/978-0387310732">978-0387310732</a>.</p>
</div>
<div id="ref-brusco_2017">
<p>71. Brusco, M.J.; Shireman, E.; Steinley, D. A comparison of latent class, k-means, and k-median methods for clustering dichotomous data. <em>Psychological methods</em> <strong>2017</strong>, <em>22</em>, 563–580.</p>
</div>
<div id="ref-felici_2007">
<p>72. Felici, G. <em>Mathematical methods for knowledge discovery and data mining</em>; Idea Group Reference: Hershey, 2007; p. 371; ISBN <a href="https://worldcat.org/isbn/978-1599045283">978-1599045283</a>.</p>
</div>
<div id="ref-Routledge_2018">
<p>73. <em>Introduction to multivariate analysis</em>; Routledge, 2018; ISBN <a href="https://worldcat.org/isbn/9780203749999">9780203749999</a>.</p>
</div>
<div id="ref-manly_2017">
<p>74. Manly, B.F. <em>Multivariate statistical methods</em>; 4th ed.; Routledge: Boca Raton, 2017; p. 270; ISBN <a href="https://worldcat.org/isbn/9781498728966">9781498728966</a>.</p>
</div>
<div id="ref-mllner_2011">
<p>75. Müllner, D. Modern hierarchical, agglomerative clustering algorithms. <em>arXiv</em> <strong>2011</strong>, <em>abs/1109.2378</em>.</p>
</div>
<div id="ref-murtagh_2012">
<p>76. Murtagh, F.; Contreras, P. Algorithms for hierarchical clustering: An overview. <em>Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery</em> <strong>2012</strong>, <em>2</em>, 86–97.</p>
</div>
<div id="ref-shaw_2003">
<p>77. Shaw, P.J.A. <em>Multivariate statistics for the environmental sciences (mathematics)</em>; 1st ed.; Hodder Education Publishers: London, 2003; p. 233; ISBN <a href="https://worldcat.org/isbn/0-340-80763-6">0-340-80763-6</a>.</p>
</div>
<div id="ref-zaslavsky_2016">
<p>78. Zaslavsky, L.; Ciufo, S.; Fedorov, B.; Tatusova, T. Clustering analysis of proteins from microbial genomes at multiple levels of resolution. <em>BMC Bioinformatics</em> <strong>2016</strong>, <em>17 Suppl 8</em>, 276.</p>
</div>
<div id="ref-hall_2011">
<p>79. Hall; Robert, D. <em>Annual plant reviews, biology of plant metabolomics</em>; 1st ed.; Wiley, John &amp; Sons, Incorporated, 2011; p. 448; ISBN <a href="https://worldcat.org/isbn/978-1-4051-9954-4">978-1-4051-9954-4</a>.</p>
</div>
<div id="ref-cai_2017">
<p>80. Cai, Q.; Alvarez, J.A.; Kang, J.; Yu, T. Network marker selection for untargeted LC-MS metabolomics data. <em>Journal of Proteome Research</em> <strong>2017</strong>, <em>16</em>, 1261–1269.</p>
</div>
<div id="ref-christin_2013">
<p>81. Christin, C.; Hoefsloot, H.C.J.; Smilde, A.K.; Hoekman, B.; Suits, F.; Bischoff, R.; Horvatovich, P. A critical assessment of feature selection methods for biomarker discovery in clinical proteomics. <em>Molecular &amp; Cellular Proteomics</em> <strong>2013</strong>, <em>12</em>, 263–276.</p>
</div>
<div id="ref-rohart_2017">
<p>82. Rohart, F.; Gautier, B.; Singh, A.; Lê Cao, K.-A. mixOmics: An r package for ’omics feature selection and multiple data integration. <em>PLoS Computational Biology</em> <strong>2017</strong>, <em>13</em>, e1005752.</p>
</div>
<div id="ref-wen_2017">
<p>83. Wen, B.; Mei, Z.; Zeng, C.; Liu, S. metaX: A flexible and comprehensive software for processing metabolomics data. <em>BMC Bioinformatics</em> <strong>2017</strong>, <em>18</em>, 183.</p>
</div>
<div id="ref-peters_2018">
<p>84. Peters, K.; Worrich, A.; Weinhold, A.; Alka, O.; Balcke, G.; Birkemeyer, C.; Bruelheide, H.; Calf, O.W.; Dietz, S.; Dührkop, K. et al. Current challenges in plant eco-metabolomics. <em>International Journal of Molecular Sciences</em> <strong>2018</strong>, <em>19</em>.</p>
</div>
<div id="ref-gromski_2014">
<p>85. Gromski, P.S.; Xu, Y.; Correa, E.; Ellis, D.I.; Turner, M.L.; Goodacre, R. A comparative investigation of modern feature selection and classification approaches for the analysis of mass spectrometry data. <em>Analytica Chimica Acta</em> <strong>2014</strong>, <em>829</em>, 1–8.</p>
</div>
<div id="ref-legendre_2012">
<p>86. Legendre, P.; Legendre, L.F.J. <em>Numerical ecology, volume 24 (developments in environmental modelling)</em>; 3rd ed.; Elsevier, 2012; p. 1006; ISBN <a href="https://worldcat.org/isbn/978-0-444-53868-0">978-0-444-53868-0</a>.</p>
</div>
<div id="ref-clarke_2009">
<p>87. Clarke, B.; Fokoue, E.; Zhang, H.H. <em>Principles and theory for data mining and machine learning</em>; Springer series in statistics; Springer New York: New York, NY, 2009; ISBN <a href="https://worldcat.org/isbn/978-0-387-98134-5">978-0-387-98134-5</a>.</p>
</div>
<div id="ref-feng_2018">
<p>88. Feng, J.; Li, B.; Jiang, X.; Yang, Y.; Wells, G.F.; Zhang, T.; Li, X. Antibiotic resistome in a large-scale healthy human gut microbiota deciphered by metagenomic and network analyses. <em>Environmental Microbiology</em> <strong>2018</strong>, <em>20</em>, 355–368.</p>
</div>
<div id="ref-fukushima_2009">
<p>89. Fukushima, A.; Kusano, M.; Redestig, H.; Arita, M.; Saito, K. Integrated omics approaches in plant systems biology. <em>Current Opinion in Chemical Biology</em> <strong>2009</strong>, <em>13</em>, 532–538.</p>
</div>
<div id="ref-vaughan_2012">
<p>90. Vaughan, A.A.; Dunn, W.B.; Allwood, J.W.; Wedge, D.C.; Blackhall, F.H.; Whetton, A.D.; Dive, C.; Goodacre, R. Liquid chromatography-mass spectrometry calibration transfer and metabolomics data fusion. <em>Analytical Chemistry</em> <strong>2012</strong>, <em>84</em>, 9848–9857.</p>
</div>
<div id="ref-wei_2018">
<p>91. Wei, R.; Wang, J.; Su, M.; Jia, E.; Chen, S.; Chen, T.; Ni, Y. Missing value imputation approach for mass spectrometry-based metabolomics data. <em>Scientific reports</em> <strong>2018</strong>, <em>8</em>, 663.</p>
</div>
<div id="ref-degenhardt_2019">
<p>92. Degenhardt, F.; Seifert, S.; Szymczak, S. Evaluation of variable selection methods for random forests and omics data sets. <em>Briefings in Bioinformatics</em> <strong>2019</strong>, <em>20</em>, 492–503.</p>
</div>
<div id="ref-saeys_2007">
<p>93. Saeys, Y.; Inza, I.; Larrañaga, P. A review of feature selection techniques in bioinformatics. <em>Bioinformatics</em> <strong>2007</strong>, <em>23</em>, 2507–2517.</p>
</div>
<div id="ref-determanjr_2014">
<p>94. Determan Jr, C.E. Optimal algorithm for metabolomics classification and feature selection varies by dataset. <em>International journal of biology</em> <strong>2014</strong>, <em>7</em>.</p>
</div>
<div id="ref-rinaudo_2016">
<p>95. Rinaudo, P.; Boudah, S.; Junot, C.; Thévenot, E.A. Biosigner: A new method for the discovery of significant molecular signatures from omics data. <em>Frontiers in molecular biosciences</em> <strong>2016</strong>, <em>3</em>, 26.</p>
</div>
<div id="ref-shi_2019">
<p>96. Shi, L.; Westerhuis, J.A.; Rosén, J.; Landberg, R.; Brunius, C. Variable selection and validation in multivariate modelling. <em>Bioinformatics</em> <strong>2019</strong>, <em>35</em>, 972–980.</p>
</div>
<div id="ref-wehrens_2012">
<p>97. Wehrens, R.; Franceschi, P. Meta-statistics for variable selection: TheR PackageBioMark. <em>Journal of statistical software</em> <strong>2012</strong>, <em>51</em>.</p>
</div>
<div id="ref-li_2013">
<p>98. Li, S.; Park, Y.; Duraisingham, S.; Strobel, F.H.; Khan, N.; Soltow, Q.A.; Jones, D.P.; Pulendran, B. Predicting network activity from high throughput metabolomics. <em>PLoS Computational Biology</em> <strong>2013</strong>, <em>9</em>, e1003123.</p>
</div>
<div id="ref-willighagen_2017">
<p>99. Willighagen, E.L.; Mayfield, J.W.; Alvarsson, J.; Berg, A.; Carlsson, L.; Jeliazkova, N.; Kuhn, S.; Pluskal, T.; Rojas-Chertó, M.; Spjuth, O. et al. The chemistry development kit (CDK) v2.0: Atom typing, depiction, molecular formulas, and substructure searching. <em>Journal of cheminformatics</em> <strong>2017</strong>, <em>9</em>, 33.</p>
</div>
<div id="ref-heller_2015">
<p>100. Heller, S.R.; McNaught, A.; Pletnev, I.; Stein, S.; Tchekhovskoi, D. Inchi, the IUPAC international chemical identifier. <em>Journal of cheminformatics</em> <strong>2015</strong>, <em>7</em>, 23.</p>
</div>
<div id="ref-backman_2011">
<p>101. Backman, T.W.H.; Cao, Y.; Girke, T. ChemMine tools: An online service for analyzing and clustering small molecules. <em>Nucleic Acids Research</em> <strong>2011</strong>, <em>39</em>, W486–91.</p>
</div>
<div id="ref-oboyle_2011">
<p>102. O’Boyle, N.M.; Banck, M.; James, C.A.; Morley, C.; Vandermeersch, T.; Hutchison, G.R. Open babel: An open chemical toolbox. <em>Journal of cheminformatics</em> <strong>2011</strong>, <em>3</em>, 33.</p>
</div>
<div id="ref-landrum_2016">
<p>103. Landrum, G. RDKit: Open-source cheminformatics software. <strong>2016</strong>.</p>
</div>
<div id="ref-kim_2016">
<p>104. Kim, S.; Thiessen, P.A.; Bolton, E.E.; Chen, J.; Fu, G.; Gindulyte, A.; Han, L.; He, J.; He, S.; Shoemaker, B.A. et al. PubChem substance and compound databases. <em>Nucleic Acids Research</em> <strong>2016</strong>, <em>44</em>, D1202–13.</p>
</div>
<div id="ref-wang_2009">
<p>105. Wang, Y.; Xiao, J.; Suzek, T.O.; Zhang, J.; Wang, J.; Bryant, S.H. PubChem: A public information system for analyzing bioactivities of small molecules. <em>Nucleic Acids Research</em> <strong>2009</strong>, <em>37</em>, W623–33.</p>
</div>
<div id="ref-pence_2010">
<p>106. Pence, H.E.; Williams, A. Chemspider: An online chemical information resource. <em>Journal of chemical education</em> <strong>2010</strong>, <em>87</em>, 1123–1124.</p>
</div>
<div id="ref-erxleben_2014">
<p>107. Erxleben, F.; Günther, M.; Krötzsch, M.; Mendez, J.; Vrandečić, D. Introducing wikidata to the linked data web. In <em>The semantic web – ISWC 2014</em>; Mika, P., Tudorache, T., Bernstein, A., Welty, C., Knoblock, C., Vrandečić, D., Groth, P., Noy, N., Janowicz, K., Goble, C., Eds.; Lecture notes in computer science; Springer International Publishing: Cham, 2014; Vol. 8796, pp. 50–65 ISBN <a href="https://worldcat.org/isbn/978-3-319-11963-2">978-3-319-11963-2</a>.</p>
</div>
<div id="ref-wohlgemuth_2010">
<p>108. Wohlgemuth, G.; Haldiya, P.K.; Willighagen, E.; Kind, T.; Fiehn, O. The chemical translation service–a web-based tool to improve standardization of metabolomic reports. <em>Bioinformatics</em> <strong>2010</strong>, <em>26</em>, 2647–2648.</p>
</div>
<div id="ref-srcinc_website_nd">
<p>109. SRC, Inc. Scientific databases.</p>
</div>
<div id="ref-ncicaddgroup_website_nd">
<p>110. Group NCI/CADD chemical identifier resolver.</p>
</div>
<div id="ref-djoumboufeunang_2016">
<p>111. Djoumbou Feunang, Y.; Eisner, R.; Knox, C.; Chepelev, L.; Hastings, J.; Owen, G.; Fahy, E.; Steinbeck, C.; Subramanian, S.; Bolton, E. et al. ClassyFire: Automated chemical classification with a comprehensive, computable taxonomy. <em>Journal of cheminformatics</em> <strong>2016</strong>, <em>8</em>, 61.</p>
</div>
<div id="ref-watrous_2012">
<p>112. Watrous, J.; Roach, P.; Alexandrov, T.; Heath, B.S.; Yang, J.Y.; Kersten, R.D.; Voort, M. van der; Pogliano, K.; Gross, H.; Raaijmakers, J.M. et al. Mass spectral molecular networking of living microbial colonies. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>2012</strong>, <em>109</em>, E1743–52.</p>
</div>
<div id="ref-kanehisa_2017">
<p>113. Kanehisa, M.; Furumichi, M.; Tanabe, M.; Sato, Y.; Morishima, K. KEGG: New perspectives on genomes, pathways, diseases and drugs. <em>Nucleic Acids Research</em> <strong>2017</strong>, <em>45</em>, D353–D361.</p>
</div>
<div id="ref-kanehisa_2019">
<p>114. Kanehisa, M.; Sato, Y.; Furumichi, M.; Morishima, K.; Tanabe, M. New approach for understanding genome variations in KEGG. <em>Nucleic Acids Research</em> <strong>2019</strong>, <em>47</em>, D590–D595.</p>
</div>
<div id="ref-kanehisa_2000">
<p>115. Kanehisa, M.; Goto, S. KEGG: Kyoto encyclopedia of genes and genomes. <em>Nucleic Acids Research</em> <strong>2000</strong>, <em>28</em>, 27–30.</p>
</div>
<div id="ref-lim_2010">
<p>116. Lim, E.; Pon, A.; Djoumbou, Y.; Knox, C.; Shrivastava, S.; Guo, A.C.; Neveu, V.; Wishart, D.S. T3DB: A comprehensively annotated database of common toxins and their targets. <em>Nucleic Acids Research</em> <strong>2010</strong>, <em>38</em>, D781–6.</p>
</div>
<div id="ref-wishart_2015">
<p>117. Wishart, D.; Arndt, D.; Pon, A.; Sajed, T.; Guo, A.C.; Djoumbou, Y.; Knox, C.; Wilson, M.; Liang, Y.; Grant, J. et al. T3DB: The toxic exposome database. <em>Nucleic Acids Research</em> <strong>2015</strong>, <em>43</em>, D928–34.</p>
</div>
<div id="ref-fahy_2007">
<p>118. Fahy, E.; Sud, M.; Cotter, D.; Subramaniam, S. LIPID MAPS online tools for lipid research. <em>Nucleic Acids Research</em> <strong>2007</strong>, <em>35</em>, W606–12.</p>
</div>
<div id="ref-shen_2019">
<p>119. Shen, X.; Wang, R.; Xiong, X.; Yin, Y.; Cai, Y.; Ma, Z.; Liu, N.; Zhu, Z.-J. Metabolic reaction network-based recursive metabolite annotation for untargeted metabolomics. <em>Nature Communications</em> <strong>2019</strong>, <em>10</em>, 1516.</p>
</div>
<div id="ref-schaefer_2009">
<p>120. Schaefer, C.F.; Anthony, K.; Krupa, S.; Buchoff, J.; Day, M.; Hannay, T.; Buetow, K.H. PID: The pathway interaction database. <em>Nucleic Acids Research</em> <strong>2009</strong>, <em>37</em>, D674–9.</p>
</div>
<div id="ref-nishimura_2001">
<p>121. Nishimura, D. BioCarta. <em>Biotech Software &amp; Internet Report</em> <strong>2001</strong>, <em>2</em>, 117–120.</p>
</div>
<div id="ref-fabregat_2018">
<p>122. Fabregat, A.; Jupe, S.; Matthews, L.; Sidiropoulos, K.; Gillespie, M.; Garapati, P.; Haw, R.; Jassal, B.; Korninger, F.; May, B. et al. The reactome pathway knowledgebase. <em>Nucleic Acids Research</em> <strong>2018</strong>, <em>46</em>, D649–D655.</p>
</div>
<div id="ref-kramer_2014">
<p>123. Kramer, F.; Bayerlová, M.; Beißbarth, T. R-based software for the integration of pathway data into bioinformatic algorithms. <em>Biology</em> <strong>2014</strong>, <em>3</em>, 85–100.</p>
</div>
<div id="ref-tenenbaum_website_2019">
<p>124. Tenenbaum, D. Bioconductor - KEGGREST 2019.</p>
</div>
<div id="ref-chang_website_2012">
<p>125. Chang, W.; Cheng, J.; Allaire, J.; Xie, Y.; McPherson, J. Shiny: Web application framework for r 2012.</p>
</div>
<div id="ref-roccaserra_2010">
<p>126. Rocca-Serra, P.; Brandizi, M.; Maguire, E.; Sklyar, N.; Taylor, C.; Begley, K.; Field, D.; Harris, S.; Hide, W.; Hofmann, O. et al. ISA software suite: Supporting standards-compliant experimental annotation and enabling curation at the community level. <em>Bioinformatics</em> <strong>2010</strong>, <em>26</em>, 2354–2356.</p>
</div>
<div id="ref-sansone_2012">
<p>127. Sansone, S.-A.; Rocca-Serra, P.; Field, D.; Maguire, E.; Taylor, C.; Hofmann, O.; Fang, H.; Neumann, S.; Tong, W.; Amaral-Zettler, L. et al. Toward interoperable bioscience data. <em>Nature Genetics</em> <strong>2012</strong>, <em>44</em>, 121–126.</p>
</div>
<div id="ref-plotlytechnologiesinc_website_2015">
<p>128. Plotly Technologies Inc Collaborative data science 2015.</p>
</div>
<div id="ref-wickham_2016">
<p>129. Wickham, H. <em>Ggplot2: Elegant graphics for data analysis</em>; Springer International Publishing : 2016; ISBN <a href="https://worldcat.org/isbn/978-3-319-24277-4">978-3-319-24277-4</a>.</p>
</div>
<div id="ref-jupyter_2018">
<p>130. Jupyter, P.; Bussonnier, M.; Forde, J.; Freeman, J.; Granger, B.; Head, T.; Holdgraf, C.; Kelley, K.; Nalvarte, G.; Osheroff, A. et al. Binder 2.0 - reproducible, interactive, sharable environments for science at scale. In Proceedings of the Proceedings of the 17th python in science conference; SciPy, 2018; pp. 113–120.</p>
</div>
<div id="ref-ram_website_nd">
<p>131. Ram, K. Configure your r project for binderhub • hole punch.</p>
</div>
<div id="ref-liggi_2018">
<p>132. Liggi, S.; Hinz, C.; Hall, Z.; Santoru, M.L.; Poddighe, S.; Fjeldsted, J.; Atzori, L.; Griffin, J.L. KniMet: A pipeline for the processing of chromatography-mass spectrometry metabolomics data. <em>Metabolomics : Official journal of the Metabolomic Society</em> <strong>2018</strong>, <em>14</em>, 52.</p>
</div>
<div id="ref-verhoeven_2018">
<p>133. Verhoeven, A.; Giera, M.; Mayboroda, O.A. KIMBLE: A versatile visual NMR metabolomics workbench in KNIME. <em>Analytica Chimica Acta</em> <strong>2018</strong>, <em>1044</em>, 66–76.</p>
</div>
<div id="ref-davidson_2016">
<p>134. Davidson, R.L.; Weber, R.J.M.; Liu, H.; Sharma-Oates, A.; Viant, M.R. Galaxy-m: A galaxy workflow for processing and analyzing direct infusion and liquid chromatography mass spectrometry-based metabolomics data. <em>GigaScience</em> <strong>2016</strong>, <em>5</em>, 10.</p>
</div>
<div id="ref-giacomoni_2015">
<p>135. Giacomoni, F.; Le Corguillé, G.; Monsoor, M.; Landi, M.; Pericard, P.; Pétéra, M.; Duperier, C.; Tremblay-Franco, M.; Martin, J.-F.; Jacob, D. et al. Workflow4Metabolomics: A collaborative research infrastructure for computational metabolomics. <em>Bioinformatics</em> <strong>2015</strong>, <em>31</em>, 1493–1495.</p>
</div>
<div id="ref-goecks_2010">
<p>136. Goecks, J.; Nekrutenko, A.; Taylor, J.; Team, G. Galaxy: A comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences. <em>Genome Biology</em> <strong>2010</strong>, <em>11</em>, R86.</p>
</div>
<div id="ref-metabohub_website_2019">
<p>137. Metabohub National infrastructure in metabolomics and fluxomics 2019.</p>
</div>
<div id="ref-guitton_2017">
<p>138. Guitton, Y.; Tremblay-Franco, M.; Le Corguillé, G.; Martin, J.-F.; Pétéra, M.; Roger-Mele, P.; Delabrière, A.; Goulitquer, S.; Monsoor, M.; Duperier, C. et al. Create, run, share, publish, and reference your LC-MS, FIA-MS, GC-MS, and NMR data analysis workflows with the Workflow4Metabolomics 3.0 galaxy online infrastructure for metabolomics. <em>The International Journal of Biochemistry &amp; Cell Biology</em> <strong>2017</strong>, <em>93</em>, 89–101.</p>
</div>
<div id="ref-workflow4metabolomics_website_nd">
<p>139. Workflow4metabolomics Referenced W4M histories workflow4metabolomics.org.</p>
</div>
<div id="ref-goble_2019">
<p>140. Goble, C.; Cohen-Boulakia, S.; Soiland-Reyes, S.; Garijo, D.; Gil, Y.; Crusoe, M.R.; Peters, K.; Schober, D. FAIR computational workflows. <em>Zenodo</em> <strong>2019</strong>.</p>
</div>
<div id="ref-wilkinson_2016">
<p>141. Wilkinson, M.D.; Dumontier, M.; Aalbersberg, I.J.J.; Appleton, G.; Axton, M.; Baak, A.; Blomberg, N.; Boiten, J.-W.; Silva Santos, L.B. da; Bourne, P.E. et al. The FAIR guiding principles for scientific data management and stewardship. <em>Scientific data</em> <strong>2016</strong>, <em>3</em>, 160018.</p>
</div>
<div id="ref-haug_2013">
<p>142. Haug, K.; Salek, R.M.; Conesa, P.; Hastings, J.; Matos, P. de; Rijnbeek, M.; Mahendraker, T.; Williams, M.; Neumann, S.; Rocca-Serra, P. et al. MetaboLights–an open-access general-purpose repository for metabolomics studies and associated meta-data. <em>Nucleic Acids Research</em> <strong>2013</strong>, <em>41</em>, D781–6.</p>
</div>
<div id="ref-sud_2016">
<p>143. Sud, M.; Fahy, E.; Cotter, D.; Azam, K.; Vadivelu, I.; Burant, C.; Edison, A.; Fiehn, O.; Higashi, R.; Nair, K.S. et al. Metabolomics workbench: An international repository for metabolomics data and metadata, metabolite standards, protocols, tutorials and training, and analysis tools. <em>Nucleic Acids Research</em> <strong>2016</strong>, <em>44</em>, D463–70.</p>
</div>
<div id="ref-gray_2017">
<p>144. Gray, A.J.G.; Goble, C.A.; Jimenez, R. Bioschemas: From potato salad to protein annotation. In Proceedings of the Proceedings of the ISWC 2017 posters &amp; demonstrations and industry tracks co-located with 16th international semantic web conference (ISWC 2017), vienna, austria, october 23rd - to - 25th, 2017.; Nikitina, N., Song, D., Fokoue, A., Haase, P., Eds.; CEUR-WS.org, 2017; Vol. 1963.</p>
</div>
<div id="ref-collberg_2016">
<p>145. Collberg, C.; Proebsting, T.A. Repeatability in computer systems research. <em>Communications of the ACM</em> <strong>2016</strong>, <em>59</em>, 62–69.</p>
</div>
<div id="ref-taschuk_2017">
<p>146. Taschuk, M.; Wilson, G. Ten simple rules for making research software more robust. <em>PLoS Computational Biology</em> <strong>2017</strong>, <em>13</em>, e1005412.</p>
</div>
<div id="ref-stanstrup_website_nd">
<p>147. Stanstrup, J.; Broeckling, C.D.; Helmus, R.; Hoffmann, N.; Mathé, E.; Naake, T.; Nicolotti, L.; Peters, K.; Rainer, J.; Salek, R. et al. The MetaRbolomics book.</p>
</div>
</div>

            </section>

          </div>
        </div>
      </div>
<a href="0300-conclusion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="9900-Appendices.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rformassspectrometry/metaRbolomics-book/edit/master/rmd/04-references/0400-references.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["The_MetaRbolomics_book.pdf", "The_MetaRbolomics_book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
