\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={The MetaRbolomics book},
            pdfauthor={Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{The MetaRbolomics book}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
    \date{}
    \predate{}\postdate{}
  
\usepackage{array}
\usepackage[table]{xcolor}
\usepackage{caption}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\newpage

\hypertarget{preface}{%
\section*{Preface}\label{preface}}
\addcontentsline{toc}{section}{Preface}

\hypertarget{how-to}{%
\subsection*{How to}\label{how-to}}
\addcontentsline{toc}{subsection}{How to}

\textbf{Download data}
The list of packages found in the tables in this book can be downloaded from
\url{public/data/AllMetaRbolomicsTables.csv}.

\textbf{Add packages to the tables}
Go to \href{https://docs.google.com/spreadsheets/d/1QTe_FhiiFEJJ7hn25V973OV3Y9up8CsiLXgZXgTg_UU/edit?usp=sharing}{the googlesheet}
and add the package. Please be careful with adding it to the right section.
If it belongs in more than one table add it mulitple times as appropiate.

The package will not appear instantly in the book but only after a change is made to
the book itself. You can also \href{https://github.com/rformassspectrometry/metaRbolomics-book/issues}{open an issue}
and request the reload such that the package shows up.

\textbf{Contributing to the text}
There are several options. In order of convenience for the maintainer you can:

\begin{itemize}
\tightlist
\item
  make a pull request on the \href{https://github.com/rformassspectrometry/metaRbolomics-book}{GitHub repository}. You will find the text in the \texttt{rmd} folder.
\item
  \href{https://github.com/rformassspectrometry/metaRbolomics-book/issues}{open an issue} with the text you want to contribute. Clearly indicate where the text belongs.
\item
  Send your contribution by email to jst( a t )nexs.ku.dk.
\end{itemize}

Remember to add yourself to author contributions.

\newpage

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

Metabolomics aims to measure, identify and (semi-)quantify a large number of metabolites in a biological system. The methods of choice are generally Nuclear Magnetic Resonance (NMR) spectroscopy or Mass Spectrometry (MS). The latter can be used directly (e.g.~direct infusion MS), but is normally coupled to a separation system such as Gas Chromatography (GC-MS), Liquid Chromatography (LC-MS) or Capillary Electrophoresis (CE-MS). In order to increase the separation power multidimensional separation systems are becoming common, such as comprehensive two-dimensional GC or LC (GC×GC, LC×LC) or LC combined with ion mobility spectrometry (LC-IMS) before MS detection. Other detection techniques include Raman spectroscopy, UV/VIS (ultraviolet/visible absorbance spectrophotometric detection- typically with a Diode Array Detector (DAD)) and fluorescence. NMR also benefits from separation techniques, such as LC-MS-NMR or LC-SPE-NMR. Additionally, there are a wide variety of pulse programs commonly used in 1D and even bigger set of 2D pulse programs used in metabolomics and for metabolite identification, for a comprehensive review on this see {[}\protect\hyperlink{ref-emwas_2019}{1}{]}. A general introduction to metabolomics can be found in textbooks like {[}\protect\hyperlink{ref-lammerhofer_2013}{2}--\protect\hyperlink{ref-wehrens_2019}{4}{]} or online courses like {[}\protect\hyperlink{ref-InternationalMetabolomicsSociety_2019}{5},\protect\hyperlink{ref-salek_website_nd}{6}{]}.

All of these analytical platforms and methodologies generate large amounts of high dimensional and complex experimental raw data when used in a metabolomics context. The amount of data, the need for reproducible research, and the complexities of the biological problem under investigation necessitates a high degree of automation and standard workflows in the data analysis. Beside vendor software, which is usually not open, open source projects offer the possibility to work in community-driven teams, perform reproducible data analysis and to work with different types of raw data. Many tools and methods have been developed to facilitate the processing and analysis of metabolomics data; most seek to solve a specific challenge in the multi-step data processing and analysis workflow.

This review provides an overview of the metabolomics-related tools that are made available as packages (and a limited number of non-trivial, non-packaged scripts) for the statistics environment and programming language R {[}\protect\hyperlink{ref-rcoredevelopmentteam_website_2018}{7}{]}. We have included packages even if they are not anymore part of current CRAN or Bioconductor, i.e.~as archived versions only. We have not included packages described in the literature if no longer available for download at all. We did include packages that are currently available, but not yet published in the scientific literature. The package descriptions have been grouped in sections according to the typical steps in the metabolomics data analysis pipeline for different analytical technologies, following the typical workflow steps from MS, NMR and UV data analysis, metabolite annotation, statistical analysis, molecular structure, network and pathway analysis and finally covering packages embracing large parts of the workflow.

\newpage

\hypertarget{metabolomics-data-processing-and-analysis}{%
\subsection{Metabolomics data processing and analysis}\label{metabolomics-data-processing-and-analysis}}

\includegraphics{figures/Figure 1.png}
\emph{Figure 1: Overview of typical tasks in metabolomics workflows, ranging from metabolite profiling (left, green) via metabolite annotation (center, purple) to data analysis using statistics and metabolite networks (right, red).}

The remainder of this section gives a broad overview and explains the typical steps, which are summarized in Figure 1, while common approaches and the available R packages are described in more detail in Section 2.

The first step for any metabolomics study is conversion from vendor formats into open data formats and pre-processing of the obtained raw data. The latter entails converting chromatographic (usually hyphenated to MS) or spectroscopic data into a data matrix suitable for data analysis. For LC-MS data this typically involves feature detection (or peak-picking) in individual samples followed by matching of features between samples. For spectroscopic data, this typically means alignment of spectra and potentially binning of the spectra into `buckets'. The final matrix will have samples in one dimension and so-called features (unique chromatographic features or spectral bins) in the other dimension. In NMR based metabolomics, several steps are carried out to process raw time domain data to a spectrum to improve quality such as phasing and baseline correction of the spectrum. Next is alignment of peaks across spectra and samples, followed by segmenting data into bins or a peak fitting step depending on the method used.

Once the analytical data has been preprocessed, it is generally subjected to different statistical approaches to find features that are ``interesting'' in the context of the experimental design, e.g.~differentiating diseased patients from healthy controls.
In untargeted metabolomics, the selected features contain only the characteristics (e.g.~m/z, retention time, chemical shift, intensity) obtained from the measurement, but not (yet) the metabolite identification or chemical structure as such. Different approaches exist for this metabolite annotation step, ranging from (usually insufficient) database lookup of exact mass (MS) or chemical shift (NMR) alone, to the use of fragmentation patterns obtained in tandem MS experiments, which can be searched against spectral databases or analysed with in silico algorithms, to spectral searching or de novo structure elucidation using combinations of NMR experiments (often 1D and 2D).

Large parts of the metabolomics software landscape in general have been covered in reviews, recent ones include the large list of software packages {[}\protect\hyperlink{ref-spicer_website_2018}{8}{]} first described by Spicer et al. {[}\protect\hyperlink{ref-spicer_2017}{9}{]}, and a series of annual reviews covering the list maintained by Misra and others {[}\protect\hyperlink{ref-misra_2016}{10}--\protect\hyperlink{ref-misra_website_2017}{13}{]}, a review by Kannan et al. {[}\protect\hyperlink{ref-kannan_2016}{14}{]} and the review focussing on approaches for compound identification of LC-MS/MS data by Blaženović et al. {[}\protect\hyperlink{ref-blaenovi_2018}{15}{]}. These reviews did include software regardless of the programming environment or language used for the implementation. In section 2.9 we briefly mention how those can be accessed from within R.

This review will focus on the ecosystem of R packages for metabolomics. It provides an overview of packages to carry out one or multiple of the above mentioned steps. Some aspects are not covered in depth or not at all. For example, MS based imaging in metabolomics is an area that has unique challenges and merits its own review, and it is also beyond the scope of this review to discuss all statistical methods that could be applied in metabolomics.

\newpage

\hypertarget{the-r-package-landscape}{%
\subsection{The R package landscape}\label{the-r-package-landscape}}

The core of the R language was started in 1997 and provided the basic functionality of a programming language, with some functions targeting statistics. The real power driving the popularity of R today is the huge number of contributed packages providing algorithms and data types for a myriad of application realms. Many packages have an Open Source license. This is not a phenomenon exclusive to R, but is rather a positive cultural aspect of bioinformatics software being mostly published under Open Source license terms, regardless of the implementation language. An R interpreter can be embedded in several other languages to execute R code snippets, and R code can also be executed via different workflow systems (e.g.~KNIME or Galaxy, see section 2.9), which is beneficial for analysis workflows, interoperability and reuse.

These packages are typically hosted on platforms that serve as an umbrella project and are a ``home'' for the developer and user communities. The Comprehensive R Archive Network (CRAN) repository contains over 14,500 packages for many application areas, including some for bioinformatics and metabolomics. The ``CRAN Task Views'', which are manually curated resources describing available packages, books etc, help users navigate CRAN and find packages for a particular task. For metabolomics, the most relevant Task View is ``\emph{Chemometrics and Computational Physics}'' {[}\protect\hyperlink{ref-mullen_website_2019}{16}{]} edited by Katharine Mullen, which includes sections on Spectroscopy, Mass Spectrometry and other tasks relevant for metabolomics applications. The Bioconductor project (BioC for short) was started by a team around Robert Gentleman in 2001 {[}\protect\hyperlink{ref-gentleman_2004}{17}{]}, and has become a vibrant community of around 1,000 contributors, working on 1,741 software, 371 data and 948 annotation packages (BioC release 3.9). In addition to a rich development infrastructure (website, developer infrastructure, version control, build farm, etc) there are regular workshops for developers and users. To enable reproducible research, BioC runs bi-annual software releases tied to a particular R release, thus ensuring and guaranteeing interoperability of packages within the same BioC release and allowing to install BioC packages from a certain release to reproduce or repeat old data analyses. On both CRAN and BioC, each package has a landing page pointing to sources, build information, binary packages and documentation. On BioC, packages are sorted (by their respective authors) into ``BiocViews'', where most packages are targeting genomics and gene expression analysis, and the most relevant ones for metabolomics are Cheminformatics (containing 11 packages), Lipidomics (11), SystemsBiology (66) and, of course, Metabolomics (56). Bioconductor workflows (organised as separate BioC View {[}\protect\hyperlink{ref-bioconductor_website_nda}{18}{]}) provide well documented examples of typical analyses. For community support, BioC maintains mailing lists, a web-based support site, slack communication channels and more. Both CRAN and BioC have a well-defined process for accepting new packages, and the respective developer guidelines (see guidelines for CRAN {[}\protect\hyperlink{ref-thecomprehensiverarchivenetwork_website_nd}{19}{]} and for BioC {[}\protect\hyperlink{ref-bioconductor_website_nd}{20}{]}) cover the package life-cycle from submission, updates and maintenance, to deprecation/orphaning of packages. In the case of BioC, new submissions undergo a peer review process, which also provides feedback on technical aspects and integration with the BioC landscape.

A smaller number of packages are also hosted on sites like \href{https://rforge.net/}{rforge.net}, \href{http://r-forge.wu-wien.ac.at/}{r-forge.wu-wien.ac.at} {[}\protect\hyperlink{ref-theul_2009}{21}{]}, or \href{https://sourceforge.net/}{sourceforge.net} (SF). The non-profit initiative rOpenSci {[}\protect\hyperlink{ref-boettiger_2015}{22}{]} maintains an ecosystem around reproducible research, including staff and community-contributed R packages with additional peer review. Currently, there are no specific metabolomics related packages.

The GitHub (and also GitLab, Bitbucket) hosting services are not specific to R development, but have gained a lot of popularity due to their excellent support for participation and contribution to software projects. The maintenance of BioC packages on one of the git-based sites has become easier since the BioC team migrated to git as its version control system. A downside of these generic repository hosting sites is that there is no central point of entry, and finding packages for specific tasks is difficult compared with dedicated platforms and relies on search engines and publications. Also, while these hosting services make it easier to provide packages that do not meet BioC and CRAN requirements (e.g.~rinchi due to limitations in the InChI algorithm itself), it also allows users to postpone (or circumvent entirely) the review process that helps ensure the quality of BioC contributions. In addition to generic search engines like \href{http://google.com}{Google}\href{http://google.com}{.com} or \href{http://bing.com}{Bing}\href{http://bing.com}{.com}, the \href{https://rdrr.io/}{rdrr.io} is a comprehensive index of R packages and documentation from CRAN, Bioconductor, GitHub and R-Forge. Initially, its main purpose was to find R packages by name, perform full-text search in package documentation, functions and R source code. Recently, it also serves as hub to actually run R code without local installation, see Section 2.9.

\newpage

\hypertarget{dependences-and-connectivity-of-metabolomics-packages}{%
\subsection{Dependences and connectivity of metabolomics packages}\label{dependences-and-connectivity-of-metabolomics-packages}}

\includegraphics{figures/Figure 2.png}
\emph{Figure 2: Dependency network of R packages. Shown in blue are packages mentioned in the review. Edges connect to packages that depend on another package, as long as they are in CRAN or BioC. Green nodes correspond to packages in CRAN or BioC not covered in the review. The inset shows the neighbourhood of the ptw package. Not shown are 1) infrastructure packages, e.g.~rJava, Rcpp 2) packages from the review without reverse dependencies and 3) data packages. Some packages from the review are not in current versions of CRAN or BioC. An interactive version of this figure is also available online (\href{https://rformassspectrometry.github.io/metaRbolomics-book/}{rformassspectrometry.github.io/metaRbolomics-book}, Appendix 2) and as supplemental file 2.}

Code reuse and object inheritance can be a sign for a well-connected and interacting community. At the useR!2015 and JSM2015 conferences, A. de Vries and J. Rickert (both Microsoft, London, UK) showed the analysis of the CRAN and BioC dependency network structure {[}\protect\hyperlink{ref-devries_website_2015b}{23}--\protect\hyperlink{ref-devries_website_2015}{25}{]}. Compared to CRAN, BioC packages had a higher connectivity: ``\emph{It seems that the Bioconductor policy encourages package authors to reuse existing material and write packages that work better together}''. We repeated such an analysis {[}\protect\hyperlink{ref-neumann_website_2019}{26}{]} with the packages mentioned in this review and created a network of reverse dependencies (i.e., the set of packages that depend on these metabolomics related packages in BioC or CRAN). The resulting network is shown in Figure 2.

\newpage

\hypertarget{r-packages-for-metabolomics}{%
\section{R-packages for metabolomics}\label{r-packages-for-metabolomics}}

\newpage

\hypertarget{mass-spectrometry-data-handling-and-pre-processing}{%
\subsection{Mass spectrometry data handling and (pre-)processing}\label{mass-spectrometry-data-handling-and-pre-processing}}

\hypertarget{profile-mode-and-centroided-data}{%
\subsubsection{Profile mode and centroided data}\label{profile-mode-and-centroided-data}}

\hypertarget{direct-infusion-mass-spectrometry-data}{%
\subsubsection{Direct infusion mass spectrometry data}\label{direct-infusion-mass-spectrometry-data}}

\hypertarget{hyphenated-ms-and-non-targeted-data}{%
\subsubsection{Hyphenated MS and non-targeted data}\label{hyphenated-ms-and-non-targeted-data}}

\hypertarget{targeted-data-and-alternative-representations-of-data}{%
\subsubsection{Targeted data and alternative representations of data}\label{targeted-data-and-alternative-representations-of-data}}

\hypertarget{additional-dimensionality}{%
\subsubsection{Additional dimensionality}\label{additional-dimensionality}}

\hypertarget{structuring-data-and-metadata}{%
\subsubsection{Structuring data and metadata}\label{structuring-data-and-metadata}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab1}R packages for mass spectrometry data handling and (pre-)processing.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab1}R packages for mass spectrometry data handling and (pre-)processing. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS data handling}}\\
Parser for common file formats: mzXML, mzData, mzML and netCDF. Usually not used directly by the end user, but provides functions to read raw data for other packages. & \href{https://doi.org/doi:10.18129/B9.bioc.mzR}{mzR} & BioC\\
Infrastructure to manipulate, process and visualise MS and proteomics data, ranging from raw to quantitative and annotated data. & \href{https://doi.org/doi:10.18129/B9.bioc.MSnbase}{MSnbase} & BioC\\
\rowcolor{gray!6}  Export and import of processed metabolomics MS results to and from the mzTab-M for metabolomics data format. & \href{https://lifs-tools.github.io/rmzTab-m/index.html}{rmzTab-M} & GitHub\\
Converts MRM-MS (.mzML) files to LC-MS style .mzML. & \href{https://github.com/wilsontom/MRMConverteR}{MRMConverteR} & GitHub\\
\rowcolor{gray!6}  Infrastructure for import, handling, representation and analysis of chromatographic MS data. & \href{https://github.com/RforMassSpectrometry/Chromatograms}{Chromatograms} & GitHub\\
Infrastructure for import, handling, representation and analysis of MS spectra. & \href{https://github.com/RforMassSpectrometry/Spectra}{Spectra} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Peak picking, grouping and alignment (LC-MS focussed or general)}}\\
Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics. & \href{https://doi.org/doi:10.18129/B9.bioc.xcms}{xcms} & BioC\\
Automatic optimization of XCMS parameters based on isotopes. & \href{http://bioconductor.org/packages/release/bioc/html/IPO.html}{IPO} & BioC\\
\rowcolor{gray!6}  Parameter tuning algorithm forXCMS, MZmine2, and other metabolomics data processingsoftware. & \href{https://doi.org/doi:10.18129/B9.bioc.Autotuner}{Autotuner} & BioC\\
Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics. & \href{http://bioconductor.org/packages/release/bioc/html/yamss.html}{yamss} & BioC\\
\rowcolor{gray!6}  Peak picking with XCMS and apLCMS, low intensity peak detection via replicate analyses. Multi-parameter feature extraction and data merging, sample quality and feature consistency evaluation. Annotation with METLIN and KEGG. & \href{https://sourceforge.net/projects/xmsanalyzer/}{xMSanalyzer} & SF\\
Pre-processing and alignment of LC-MS data without assuming a parametric peak shape model allowing maximum flexibility. It utilizes the knowledge of known metabolites, as well as robust machine learning. & \href{http://web1.sph.emory.edu/apLCMS/}{apLCMS} & SF\\
\rowcolor{gray!6}  Peak detection using chromatogram subregion detection, consensus integration bound determination and Accurate missing value integration. Outputs in XCMS-compatible format. & \href{https://github.com/nathaniel-mahieu/warpgroup}{warpgroup} & GitHub\\
Peak picking for (LC/GC-)MS data, improving the detection of low abundance signals via a master map of m/z/RT space before peak detection. Results are XCMS-compatible. & \href{https://doi.org/doi:10.18129/B9.bioc.cosmiq}{cosmiq} & BioC\\
\rowcolor{gray!6}  m/z detection (i.e. peak-picking) for accurate mass data, collecting all data points above an intensity threshold, grouping them by m/z values and estimating representative m/z values for the clusters; extracting EICs. & \href{http://amdorap.sourceforge.net/}{AMDORAP} & SF\\
(GC/LC)-MS data analysis for environmental science, including raw data processing, analysis of molecular isotope ratios, matrix effects, and short-chain chlorinated paraffins. & \href{https://cran.r-project.org/web/packages/enviGCMS/index.html}{enviGCMS} & CRAN\\
\rowcolor{gray!6}  Sequential partitioning, clustering and peak detection of centroided LC-MS mass spectrometry data (.mzXML), with Interactive result and raw data plot. & \href{https://cran.r-project.org/web/packages/enviPick/index.html}{enviPick} & CRAN\\
PeakpickingwithXCMS. Groups chemically related features beforealignmentacross samples. Additional processing after alignment includes feature validation, re-integration and annotation based on custom database. & \href{https://github.com/lauzikaite/massFlowR}{massFlowR} & GitHub\\
\rowcolor{gray!6}  KPIC2 extracts pure ion chromatograms (PIC) via K-means clustering of ions in region of interest, performs grouping and alignment, grouping of isotopic and adduct features, peak filling and Random Forest classification. & \href{https://github.com/hcji/KPIC2}{KPIC2} & GitHub\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Isotope labeling using MS}}\\
Analysis of untargeted LC/MS data from stable isotope-labeling experiments. Also uses XCMS for feature detection. & \href{https://github.com/jcapelladesto/geoRge}{geoRge} & GitHub\\
\rowcolor{gray!6}  Correction of MS and MS/MS data from stable isotope labeling (any tracer isotope) experiments for natural isotope abundance and tracer impurity. Separate GUI available in IsoCorrectoRGUI. & \href{https://bioconductor.org/packages/release/bioc/html/IsoCorrectoR.html}{IsoCorrectoR} & BioC\\
Extension of XCMS that provides support for isotopic labeling. Detection of metabolites that have been enriched with isotopic labeling. & \href{http://pattilab.wustl.edu/software/x13cms/x13cms.php}{X13CMS} & NA\\
\rowcolor{gray!6}  Analysis of isotopic patterns in isotopically-labeled MS data. Estimates the isotopic abundance of the stable isotope (either 2H or 13C) within specified compounds. & \href{https://github.com/RuggeroFerrazza/IsotopicLabelling}{IsotopicLabelling} & GitHub\\
Finding the dual (or multiple) isotope labeled analytes using dual labeling of metabolites for metabolome analysis (DLEMMA) approach, described in Liron [42]. & \href{https://cran.r-project.org/web/packages/Miso}{Miso} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Targeted MS}}\\
Peak picking using peak apex intensities for selected masses. Reference library matching, RT/RI conversion plus metabolite identification using multiple correlated masses. Includes GUI. & \href{http://bioconductor.org/packages/2.5/bioc/html/TargetSearch.html}{TargetSearch} & BioC\\
Pre-processing for targeted (SIM) GC-MS data. Guided selection of appropriate fragments for the targets of interest by using an optimization algorithm based on user provided library. & \href{https://doi.org/doi:10.18129/B9.bioc.SIMAT}{SIMAT} & BioC\\
\rowcolor{gray!6}  Deconvolution of MS2 spectra obtained with wide isolation windows. & \href{http://pattilab.wustl.edu/software/decoms2/decoms2.php}{decoMS2} & NA\\
Deconvolution of SWATH-MS experiments to MRM transitions. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=128}{SWATHtoMRM} & NA\\
\rowcolor{gray!6}  Automatic analysis of large scale MRM experiments. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=34}{MRMAnalyzer} & NA\\
Tailors peak detection for targeted metabolites through iterative user interface. It automatically integrates peak areas for all isotopologues and outputs extracted ion chromatograms (EICs). & \href{https://github.com/stevehoang/assayr}{AssayR} & GitHub\\
\rowcolor{gray!6}  Targeted peak picking and annotation. Includes Shiny GUI. & \href{https://github.com/phenomecentre/peakPantheR}{peakPantheR} & GitHub\\
Toolkit for working with Selective Reaction Monitoring (SRM) MS data and other variants of targeted LC-MS data. & \href{https://github.com/wilsontom/sRm}{sRm} & GitHub\\
\rowcolor{gray!6}  Deconvolution of SWATH-MS data. & \href{https://github.com/ZhuMSLab/DecoMetDIA}{DecoMetDIA} & GitHub\\
Targeted peak picking and annotation. All functions through Shiny GUI. & \href{https://github.com/hcji/TarMet}{TarMet} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{GC-MS and GC×GC-MS}}\\
Unsupervised data mining on GC-MS. Clustering of mass spectra to detect compound spectra. The output can be searched in NIST and ARISTO [50]. & \href{https://cran.r-project.org/package=MSeasy}{MSeasy} & CRAN\\
Pre-processing for GC/MS, MassBank search, NIST format export. & \href{https://cran.r-project.org/package=erah}{erah} & CRAN\\
\rowcolor{gray!6}  Pre-processing using AMDIS [53, 54] for untargeted GC-MS analysis. Feature grouping across samples, improved quantification, removal of false positives, normalisation via internal standard or biomass; basic statistics. & \href{https://doi.org/doi:10.18129/B9.bioc.Metab}{Metab} & BioC\\
Deconvolution of GC-MS and GC×GC-MS unit resolution data using orthogonal signal deconvolution (OSD), independent component regression (ICR) and multivariate curve resolution (MCR-ALS). & \href{http://cran.r-project.org/package=osd}{osd} & CRAN\\
\rowcolor{gray!6}  Corrects overloaded signals directly in raw data (from GC-APCI-MS) automatically by using a Gaussian or isotopic-ratio approach. & \href{https://cran.r-project.org/package=CorrectOverloadedPeaks}{CorrectOverloadedPeaks} & CRAN\\
Alignment of GC data. Also GC-FID or any single channel data since it works directly on peak lists. & \href{https://cran.r-project.org/package=GCalignR}{GCalignR} & CRAN\\
\rowcolor{gray!6}  GC-MS data processing and compound annotation pipeline. Includes the building, validating, and query of in-house databases. & \href{https://www.bioconductor.org/packages/release/bioc/html/metaMS.html}{metaMS} & BioC\\
Peak picking for GC×GC-MS using bayes factor and mixture probability models. & \href{http://mrr.sourceforge.net/download.html}{msPeak} & SF\\
\rowcolor{gray!6}  Peak alignment for GC×GC-MS data with homogeneous peaks based on mixture similarity measures. & \href{http://mrr.sourceforge.net/download.html}{mSPA} & SF\\
Peak alignment for GC×GC-MS data with homogeneous and/or heterogenous peaks based on mixture similarity measures. & \href{http://mrr.sourceforge.net/download.html}{SWPA} & SF\\
\rowcolor{gray!6}  Chemometrics analysis GC×GC-MS: baseline correction, smoothing, COW peak alignment, multiway PCA is incorporated. & \href{https://cran.r-project.org/package=RGCxGC}{RGCxGC} & CRAN\\
Retention time and mass spectra similarity threshold-free alignments, seamlessly integrates retention time standards for universally reproducible alignments, performs common ion filtering, and provides compatibility with multiple peak quantification methods. & \href{https://github.com/rramaker/R2DGC}{R2DGC} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Flow injection / direct infusion analysis}}\\
Pre-processing of data from Flow Injection Analysis (FIA) coupled to High-Resolution Mass Spectrometry (HRMS). & \href{https://doi.org/doi:10.18129/B9.bioc.proFIA}{proFIA} & BioC\\
Flow In-jection Electrospray Mass Spectrometry Processing: data processing, classification modelling and variable selection in metabolite fingerprinting & \href{https://github.com/aberHRML/FIEmspro}{FIEmspro} & GitHub\\
\rowcolor{gray!6}  Processing Mass Spectrometry spectrum by using wavelet based algorithm. Can be used for direct infusion experiments. & \href{http://bioconductor.org/packages/release/bioc/html/MassSpecWavelet.html}{MassSpecWavelet} & BioC\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Other}}\\
Filtering of features originating from artifactual interference. Based on the analysis of an extract of E. coli grown in 13C-enriched media. & \href{https://github.com/pattilab/credential}{credential} & GitHub\\
\rowcolor{gray!6}  Wrappers for XCMS and CAMERA. Also includes matching to a spectral library and a GUI. & \href{https://doi.org/doi:10.18129/B9.bioc.metaMS}{metaMS} & BioC\\
Processing of peaktables from AMDIS, XCMS or ChromaTOF. Functions for plotting also provided. & \href{https://doi.org/doi:10.18129/B9.bioc.flagme}{flagme} & BioC\\
\rowcolor{gray!6}  Parametric Time Warping (RT correction) for both DAD and LC-MS. & \href{https://cran.r-project.org/package=ptw}{ptw} & CRAN\\
R wrapper for X!Tandem software for protein identification. & \href{http://bioconductor.org/packages/release/bioc/html/rTANDEM.html}{rTANDEM} & BioC\\
\rowcolor{gray!6}  Building, validation, and statistical analysis of extended assay libraries for SWATH proteomics data. & \href{https://bioconductor.org/packages/release/bioc/html/SwathXtend.html}{SwathXtend} & BioC\\
Split a data set into a set of likely true metabolites and likely measurement artifacts by comparing missing rates of pooled plasma samples and biological samples. & \href{https://cran.r-project.org/package=MetProc}{MetProc} & CRAN\\
\rowcolor{gray!6}  Quality of LC-MS and direct infusion MS data. Generates a report that contains a comprehensive set of quality control metrics and charts. & \href{https://github.com/computational-metabolomics/qcrms}{qcrms} & GitHub\\*
\end{longtable}

\hypertarget{ion-species-grouping-and-annotation}{%
\subsubsection{Ion species grouping and annotation}\label{ion-species-grouping-and-annotation}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab2}R packages for ion species grouping, annotation, molecular formula generation and accurate mass lookup.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab2}R packages for ion species grouping, annotation, molecular formula generation and accurate mass lookup. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Molecular formula and isotope analysis}}\\
Simulation of and decomposition of Isotopic Patterns. & \href{http://bioconductor.org/packages/release/bioc/html/Rdisop.html}{Rdisop} & BioC\\
Calculation of isotope fine patterns. Also adduct calculations and molecular formula parsing. Web version available at www.envipat.eawag.ch. & \href{https://cran.r-project.org/package=enviPat}{enviPat} & CRAN\\
\rowcolor{gray!6}  Molecular formula assignment, mass recalibration, signal-to-noise evaluation, and unambiguous formula selections are provided. & \href{https://github.com/ChARM-Group/MFAssignR}{MFAssignR} & GitHub\\
Uses GenForm for molecular formula generation on mass accuracy, isotope and/or MS/MS fragments, as well as performing MS/MS subformula annotation. & \href{https://github.com/schymane/GenFormR}{GenFormR} & GitHub\\
\rowcolor{gray!6}  Checking element isotopes, calculating (isotope labelled) exact monoisotopic mass,m/zvalues, mass accuracy, and inspecting possible contaminant mass peaks, examining possible adducts in ESI and MALDI. & \href{https://cran.r-project.org/package=MSbox}{MSbox} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS feature grouping}}\\
Grouping of correlated features into pseudo compound spectra using correlation across samples and similarity of peak shape. Annotation of isotopes and adducts. Works as an add-on to XCMS. & \href{https://doi.org/doi:10.18129/B9.bioc.CAMERA}{CAMERA} & BioC\\
\rowcolor{gray!6}  Grouping of features based on similarity between coelution profiles. & \href{https://cran.r-project.org/web/packages/cliqueMS/}{CliqueMS} & CRAN\\
Cluster based feature grouping for non-targeted GC or LC-MS data. & \href{https://github.com/cbroeckl/RAMClustR}{RAMClustR} & CRAN\\
\rowcolor{gray!6}  Uses dynamic block summarisation to group features belong to the same compound. Correction for peak misalignments and isotopic pattern validation. & \href{https://sourceforge.net/projects/mettailor/}{MetTailor} & SF\\
Isotope {\&} adduct peak grouping, homologous series detection. & \href{https://cran.r-project.org/web/packages/nontarget/}{nontarget} & CRAN\\
\rowcolor{gray!6}  Bayesian approach for grouping peaks originating from the same compound. & \href{http://research.cs.aalto.fi/pml/software/peakANOVA/}{peakANOVA} & NA\\
Combination of data from positive and negative ionization mode finding common molecular entities. & \href{https://cran.r-project.org/package=MScombine}{MScombine} & CRAN\\
\rowcolor{gray!6}  Grouping of correlated features into pseudo compound spectra using correlation across sample. Annotation of isotopes and adducts. Can work directly with the XCMS output. & \href{http://www.urr.cat/AStream/AStream.html}{Astream} & NA\\
Navigation of high-resolution MS/MS data in a GUI based on mass spectral similarity. & \href{https://bioconductor.org/packages/release/bioc/html/MetCirc.html}{MetCirc} & BioC\\
\rowcolor{gray!6}  Deconvolution of MS/MS spectra obtained with wide isolation windows. & \href{http://pattilab.wustl.edu/software/decoms2/decoms2.php}{decoMS2} & NA\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Ion/adduct/fragment annotation}}\\
Bayesian probabilistic annotation. & \href{https://github.com/rsilvabioinfo/ProbMetab}{ProbMetab} & GitHub\\
\rowcolor{gray!6}  Isotope {\&} adduct peak grouping, unsupervised homologous series detection. & \href{https://cran.r-project.org/web/packages/nontarget/}{nontarget} & CRAN\\
Automatic interpretation of fragments and adducts in MS spectra. Molecular formula prediction based on fragmentation. & \href{https://cran.r-project.org/package=InterpretMSSpectrum}{InterpretMSSpectrum} & CRAN\\
\rowcolor{gray!6}  Automated annotation using MS2 data or databases and retention time. Calculation of spectral and chemical networks. & \href{https://github.com/WMBEdmands/compMS2Miner}{compMS2Miner} & GitHub\\
Screening, annotation, and putative identification of mass spectral features in lipidomics. Default databases contain {\char`\~}25,000 compounds. & \href{https://doi.org/doi:10.18129/B9.bioc.LOBSTAHS}{LOBSTAHS} & BioC\\
\rowcolor{gray!6}  Automated annotation of fragments from MS and MS2 and putative identification against simulated library fragments of {\char`\~}500,000 lipid species across {\char`\~}60 lipid types. & \href{https://github.com/GarrettLab-UF/LipidMatch}{LipidMatch} & GitHub\\
Annotation of lipid type and acyl groups on independent acquisition-mass spectrometry lipidomics based on fragmentation and intensity rules. & \href{https://cran.r-project.org/package=LipidMS}{LipidMS} & CRAN\\
\rowcolor{gray!6}  Accurate mass and/or retention time and/or collisional cross section matching. & \href{https://github.com/michaelwitting/masstrixR}{masstrixR} & GitHub\\
Downloads KEGG compounds orthology data and wraps the KEGGREST package to extractgene data. & \href{https://cran.r-project.org/package=omu}{omu} & CRAN\\
\rowcolor{gray!6}  Paired mass distance analysis to find independent peaks inm/z-retention time profiles based on retention time hierarchical cluster analysis and frequency analysis of paired mass distances within retention time groups. Structure directed analysis to find potential relationship among those independent peaks. Shiny GUI included. & \href{https://cran.r-project.org/package=pmd}{pmd} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{metabolite-identification-with-msms-data}{%
\subsection{Metabolite identification with MS/MS data}\label{metabolite-identification-with-msms-data}}

\hypertarget{msms-data-handling-spectral-matching-and-clustering}{%
\subsubsection{MS/MS data handling, spectral matching and clustering}\label{msms-data-handling-spectral-matching-and-clustering}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab3}R packages for MS/MS data.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab3}R packages for MS/MS data. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS2 and libraries}}\\
Tools for processing raw data to database ready cleaned spectra with metadata. & \href{https://www.bioconductor.org/packages/release/bioc/html/RMassBank.html}{RMassBank} & BioC\\
From RT-m/z pairs (or m/z alone) creates MS2 experiment files with non-overlapping subsets of the targets. Bruker, Agilent and Waters supported. & \href{https://github.com/sneumann/MetShot}{MetShot} & GitHub\\
\rowcolor{gray!6}  Creating MS libraries from LC-MS data using XCMS/CAMERA packages. A multi-modular annotation function including X-Rank spectral scoring matches experimental data against the generated MS library. & \href{https://github.com/AharoniLab/MatchWeiz}{MatchWeiz} & GitHub\\
Assess precursor contribution to fragment spectrum acquired or anticipated isolation windows using "precursor purity" for both LC-MS(/MS) and DI-MS(/MS) data. Spectral matching against a SQLite database of library spectra. & \href{https://doi.org/doi:10.18129/B9.bioc.msPurity}{msPurity} & BioC\\
\rowcolor{gray!6}  Automated quantification of metabolites by targeting mass spectral/retention time libraries into full scan-acquired GC-MS chromatograms. & \href{https://cran.r-project.org/package=baitmet}{baitmet} & CRAN\\
MS2 spectra similarity and unsupervised statistical methods. Workflow from raw data to visualisations and is interfaceable with XCMS. & \href{https://bioconductor.org/packages/CluMSID/}{CluMSID} & BioC\\
\rowcolor{gray!6}  Import of spectra from different file formats such as NIST msp, mgf (mascot generic format), and library (Bruker) to MSnbase objects. & \href{https://github.com/meowcat/MSnio}{MSnio} & GitHub\\
Multi-purpose mass spectrometry package. Contains many different functions .e.g. isotope pattern calculation, spectrum similarity, chromatogram plotting, reading of msp files and peptide related functions. & \href{https://cran.r-project.org/package=OrgMassSpecR}{OrgMassSpecR} & CRAN\\
\rowcolor{gray!6}  Annotation of LC-MS data based on a database of fragments. & \href{https://cran.r-project.org/package=MetaboList}{MetaboList} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{In silico fragmentation}}\\
In silico fragmentation of candidate structures. & \href{https://github.com/c-ruttkies/MetFragR}{MetFragR} & GitHub\\
\rowcolor{gray!6}  SOLUTIONS for High ReSOLUTION Mass Spectrometry including several functions to interact with MetFrag, developed during the SOLUTIONS project (www.solutions-project.eu). & \href{https://github.com/schymane/ReSOLUTION}{ReSOLUTION} & GitHub\\
Uses MetFrag and adds substructure prediction using the isotopic pattern. Can be trained on a custom dataset. & \href{https://github.com/lucanard/CCC}{CCC} & GitHub\\
\rowcolor{gray!6}  Retention time prediction based on compound structuredescriptors. Five different machine learning algorithms are available to build models. Plotting available to explore chemical space and model quality assessment. & \href{https://github.com/PaoloBnn/Retip}{Retip} & GitHub\\*
\end{longtable}

\hypertarget{reading-of-spectral-databases}{%
\subsubsection{Reading of spectral databases}\label{reading-of-spectral-databases}}

\newpage

\hypertarget{nmr-data-handling-and-pre-processing}{%
\subsection{NMR data handling and (pre-)processing}\label{nmr-data-handling-and-pre-processing}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab4}R packages for NMR data handling, (pre-)processing and analysis.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab4}R packages for NMR data handling, (pre-)processing and analysis. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Data processing and Analysis}}\\
A tool for processing of 1H NMR data including: Apodization, baseline correction, bucketing, Fourier transformation, warping and phase correction. Bruker FID can be directly imported. & \href{https://github.com/ManonMartin/SOAP-NMR}{PepsNMR} & GitHub\\
Spectra alignment, peak picking based processing, Quantitative analysis and visualizations for 1D NMR. & \href{https://cran.r-project.org/package=speaq}{speaq} & CRAN\\
\rowcolor{gray!6}  Interactive environment based on R-Shiny that includes a complete set of tools to process and visualize 1D NMR spectral data. Processing includes baseline correction, ppm calibration, removal of solvents and contaminants and re-alignment of chemical shifts. & \href{https://www.nmrprocflow.org/}{NMRProcFlow} & Bitbucket\\
TheMetaboMateR toolbox covers basic processing andstatisticalanalysis steps including; several spectral quality assessment (such as dealing with baseline distortions, water suppression to quality assessment of shimming and line width) as well as pre-processing (referencing,baselinecorrection, ... ) to multivariate analysis statistics functions. & \href{https://github.com/kimsche/MetaboMate}{MetaboMate} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Data Analysis and Identification}}\\
Analysis of 1D and 2D NMR spectra using a ROIs based approach. Export to MMCD or uploaded to BMRB for identification. & \href{http://rnmr.nmrfam.wisc.edu/}{rNMR} & NA\\
Pre-processing and identification in an R-based GUI for 1D NMR. & \href{https://github.com/danielcanueto/rDolphin}{rDolphin} & GitHub\\
\rowcolor{gray!6}  Bayesian automated metabolite analyser for 1D NMR spectra. Deconvolution of NMR spectra and automate metabolite quantification. Also identification based on chemical shift lists. & \href{http://batman.r-forge.r-project.org/}{BATMAN} & RF\\
“ASICS: an automatic method for identification and quantification of metabolites in complex 1D 1H NMR spectra.” & \href{https://bioconductor.org/packages/release/bioc/html/ASICS.html}{ASICS} & BioC\\
\rowcolor{gray!6}  ASICSdata: 1D NMR spectra for ASICS. & \href{https://www.bioconductor.org/packages/release/data/experiment/vignettes/ASICSdata/inst/doc/ASICSdata.html}{ASICSdata} & BioC\\
shiny-based interactive NMR data import and Statistical TOtal Correlation SpectroscopY (STOCSY) analyses. & \href{https://cran.r-project.org/package=iSTATS}{iSTATS} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{NMR and integration with Genomics}}\\
MWASTools: an integrated pipeline to perform NMR based metabolome-wide association studies (MWAS). Quality control analysis; MWAS using various models (partial correlations, generalized linear models); visualization of statistical outcomes; metabolite assignment using STOCSY; and biological interpretation of MWAS results. & \href{https://bioconductor.org/packages/release/bioc/html/MWASTools.html}{MWASTools} & BioC\\
An Integrated Suite for Genetic Mapping of Quantitative Variations of 1H NMR-Based Metabolic Profiles. mQTL-NMR provides a complete metabotype quantitative trait locus (mQTL) mapping analysis pipeline for metabolomic data. & \href{NA}{mQTL.NMR} & BioC\\
\rowcolor{gray!6}  Handles hyperspectral data, i.e. spectra plus further information such as spatial information, time, concentrations, etc. Such data are frequently encountered in Raman, IR, NIR, UV/VIS, NMR, MS, etc. & \href{https://cran.r-project.org/package=ChemoSpec}{hyperSpec} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{uv-data-handling-and-pre-processing}{%
\subsection{UV data handling and (pre-)processing}\label{uv-data-handling-and-pre-processing}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab5}R Packages for UV data handling and (pre-)processing.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab5}R Packages for UV data handling and (pre-)processing. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{DAD}}\\
Multivariate Curve Resolution (Alternating Least Squares) for DAD data. & \href{https://github.com/rwehrens/alsace}{alsace} & GitHub\\
Parametric Time Warping (RT correction) for both DAD and LC-MS. & \href{https://cran.r-project.org/package=ptw}{ptw} & CRAN\\
\rowcolor{gray!6}  Handles hyperspectral data, i.e. spectra plus further information such as spatial information, time, concentrations, etc. Such data are frequently encountered in Raman, IR, NIR, UV/VIS, NMR, MS, etc. & \href{https://cran.r-project.org/package=ChemoSpec}{hyperSpec} & CRAN\\
Projection based methods for preprocessing, exploring and analysis of multivariate data. & \href{https://cran.r-project.org/package=mdatools}{mdatools} & CRAN\\
\rowcolor{gray!6}  Collection of baseline correction algorithms, along with a GUI for optimising baseline algorithm parameters. & \href{https://cran.r-project.org/package=baseline}{baseline} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{statistical-analysis-of-metabolomics-data}{%
\subsection{Statistical analysis of metabolomics data}\label{statistical-analysis-of-metabolomics-data}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab6}R packages for statistical analysis of metabolomics data.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab6}R packages for statistical analysis of metabolomics data. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Sample size}}\\
Estimate sample sizes for metabolomics experiments, (NMR and targeted approaches supported). & \href{https://cran.r-project.org/package=MetSizeR}{MetSizeR} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Normalization}}\\
Cross-contribution robust multiple standard normalization. Normalization using internal standards. & \href{https://cran.r-project.org/web/packages/crmn/index.html}{crmn} & CRAN\\
\rowcolor{gray!6}  Within and between batch correction of LC-MS metabolomics data using either QC samples or all samples. & \href{https://gitlab.com/CarlBrunius/batchCorr}{batchCorr} & GitLab\\
Normalisation for low concentration metabolites. Mixed model with simultaneous estimation of a correlation matrix. & \href{https://sourceforge.net/projects/metabnorm/}{Metabnorm} & SF\\
\rowcolor{gray!6}  A collection of data distribution normalization methods. & \href{http://quantitativeproteomics.org/normalyzer}{Normalizer} & NA\\
Functions for drift removal and data normalisation based on: component correction, median fold change, ComBat or common PCA (CPCA). & \href{http://b2slab.upc.edu/software-and-downloads/intensity-drift-correction/}{intCor} & NA\\
\rowcolor{gray!6}  Normalisation using a singular value decomposition. & \href{https://sourceforge.net/projects/eigenms/}{EigenMS} & SF\\
Normalization based on RUV-random [164]. & \href{https://cran.r-project.org/web/packages/MetNorm/index.html}{MetNorm} & CRAN\\
\rowcolor{gray!6}  SVR-based normalization and integration for large-scale metabolomics data. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=39}{MetNormalizer} & GitHub\\
Drift correction using QC samples or all study samples. & \href{https://github.com/rwehrens/BatchCorrMetabolomics}{BatchCorrMetabolomics} & GitHub\\
\rowcolor{gray!6}  Signal and Batch Correction for Mass Spectrometry & \href{https://github.com/computational-metabolomics/sbcms}{SBCMS} & GitHub\\
Multiple fitting models to correct intra- and inter-batch effects. & \href{https://cran.r-project.org/package=MetaboQC}{MetaboQC} & CRAN\\
\rowcolor{gray!6}  Collection of functions designed to implement, assess, and choose a suitable normalization method for a given metabolomics study. & \href{https://cran.r-project.org/package=NormalizeMets}{NormalizeMets} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Exploratory Data Analysis}}\\
A large number of methods available for PCA. & \href{http://bioconductor.org/packages/release/bioc/html/pcaMethods.html}{pcaMethods} & BioC\\
\rowcolor{gray!6}  Chemometric analysis of NMR, IR or Raman spectroscopy data. It includes functions for spectral visualisation, peak alignment, HCA, PCA and model-based clustering. & \href{https://cran.r-project.org/package=ChemoSpec}{ChemoSpec} & BioC\\
Joint analysis of MS and MS2 data, where hierarchical cluster analysis is applied to MS2 data to annotate metabolite families and principal component analysis is applied to MS data to discover regulated metabolite families. & \href{https://github.com/ipb-halle/MetFamily}{MetFamily} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Univariate hypothesis testing}}\\
Many methods for corrections for multiple testing. & \href{http://bioconductor.org/packages/release/bioc/html/multtest.html}{multtest} & BioC\\
Estimate tail area-based false discovery rates (FDR) as well as local false discovery rates (fdr) for a variety of null models (p-values, z-scores, correlation coefficients, t-scores). & \href{http://strimmerlab.org/software/fdrtool/}{fdrtool} & CRAN\\
\rowcolor{gray!6}  GUI for statistical analysis using linear mixed models to normalize data and ANOVA to test for treatment effects. & \href{http://metabr.r-forge.r-project.org/}{MetabR} & RF\\
Derives stable estimates of the metabolome-wide significance level within a univariate approach based on a permutation procedure which effectively controls the maximum overall type I error rate at the α level. & \href{https://github.com/AlinaPeluso/MWSL}{MWSL} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Multivariate modeling and feature selection}}\\
Various multivariate methods to analyze metabolomics datasets. Main methods include PCA, Partial Least Squares regression (PLS), and extensions to the PLS like PLS Discriminant Analysis PLS-DA and the orthogonal variants OPLS(-DA). & \href{http://bioconductor.org/packages/release/bioc/html/ropls.html}{ropls} & BioC\\
Package for performing Partial Least Squares regression (PLS). & \href{https://cran.r-project.org/package=pls}{pls} & CRAN\\
\rowcolor{gray!6}  PPCA, PPCCA, MPPCA. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
General framework for building regression and classification models. & \href{https://cran.r-project.org/package=caret}{caret} & CRAN\\
\rowcolor{gray!6}  ASCA, figure of merit (FoM), PCA, Goeman’s global test for metabolomic pathways (Q-stat), Penalized Jacobian method (for calculating network connections), Time-lagged correlation method and zero slopes method. It also includes centering and scaling functions. & \href{https://cran.r-project.org/package=MetStaT}{MetStaT} & CRAN\\
RF for the construction, optimization and validation of classification models with the aim of identifying biomarkers. Also normalization, scaling, PCA, MDS. & \href{https://cran.r-project.org/package=RFmarkerDetector}{RFmarkerDetector} & CRAN\\
\rowcolor{gray!6}  PLS-DA, RF, SVM, GBM, GLMNET, PAM. & \href{http://bioconductor.org/packages/release/bioc/html/OmicsMarkeR.html}{OmicsMarkeR} & BioC\\
Recursive feature elimination approach that selects features which significantly contribute to the performance of PLS-DA, Random Forest or SVM classifiers. & \href{http://bioconductor.org/packages/release/bioc/html/biosigner.html}{biosigner} & BioC\\
\rowcolor{gray!6}  Find Biomarkers in two class discrimination problems with variable selection methods provided for several classification metods (lasso/elastic net, PC-LDA, PLS-DA, and several t-tests). & \href{https://cran.r-project.org/web/packages/BioMark/index.html}{BioMark} & CRAN\\
Unsupervised feature extraction specifically designed for analysing noisy and high-dimensional datasets. & \href{https://cran.r-project.org/package=KODAMA}{KODAMA} & CRAN\\
\rowcolor{gray!6}  Non-parametric method for identifying differentially expressed features based on the estimated percentage of false predictions. & \href{https://doi.org/doi:10.18129/B9.bioc.RankProd}{RankProd} & CRAN\\
Fits multi-way component models via alternating least squares algorithms with optional constraints: orthogonal, non-negative, unimodal, monotonic, periodic, smooth, or structure. Fit models include InDScal, PARAFAC, PARAFAC2, SCA, Tucker. & \href{https://cran.r-project.org/package=multiway}{multiway} & CRAN\\
\rowcolor{gray!6}  Decompose a tensor of any order, as a generalisation of SVD also supporting non-identity metrics and penalisations. 2-way SVD is also available. Also includes PCAn (Tucker-n) and PARAFAC/CANDECOMP. & \href{https://cran.r-project.org/package=PTAk}{PTAk} & CRAN\\
Fits multi-way component models via alternating least squares algorithms with optional constraints. Fit models include Individual Differences Scaling, Multiway Covariates Regression, PARAFAC (1 and 2), SCA, and Tucker Factor Analysis. & \href{https://cran.r-project.org/package=ThreeWay}{ThreeWay} & CRAN\\
\rowcolor{gray!6}  Performs variable selection in a multivariate linear model by estimating the covariance matrix of the residuals then use it to remove the dependence that may exist among the responses and eventually performs variable selection by using the Lasso criterion. & \href{https://cran.r-project.org/package=MultiVarSel}{MultiVarSel} & CRAN\\
Performs the O2PLS data integration method for two datasets yielding joint and data-specific parts for each dataset. & \href{https://cran.r-project.org/package=OmicsPLS}{OmicsPLS} & CRAN\\
\rowcolor{gray!6}  Contains ordination methods such as ReDundancy Analysis (RDA), (Canonical or Detrended) Correspondence Analysis (CCA, DCA for binary explanatory variables), (Non-metric) Multi-Dimensional Scaling ((N)MDS) and other univariate and multivariate methods. Originally developed for vegetation ecologists, many functions are also applicable to metabolomics. & \href{https://cran.r-project.org/package=vegan}{vegan} & CRAN\\
Linear and non-linear Discriminant Analysis methods (e.g. LDA), stepwise selection and classification methods useful for feature selection. & \href{https://cran.r-project.org/package=klaR}{klaR} & CRAN\\
\rowcolor{gray!6}  Variable selection methods for PLS, including significance multivariate correlation (SMC), selectivity ratio (SR), variable importance in projections (VIP), loading weights (LW), and regression coefficients (RC). It contains also some other modelling methods. & \href{https://cran.r-project.org/package=plsVarSel}{plsVarSel} & CRAN\\
Predictive multivariate modelling using PLS and Random Forest Data. Repeated double cross unbiased validation and variable selection. & \href{https://gitlab.com/CarlBrunius/MUVR}{MUVR} & GitLab\\
\rowcolor{gray!6}  Biomarker validation for predicting survival. Cross validation methods to validate and select biomarkers when the outcome of interest is survival. & \href{https://cran.r-project.org/package=MetabolicSurv}{MetabolicSurv} & CRAN\\
Pre-treatment, classification, feature selection and correlation analyses of metabolomics data. & \href{https://github.com/jasenfinch/metabolyseR}{metabolyseR} & GitHub\\
\rowcolor{gray!6}  Components search, optimal model components number search, optimal model validity test by permutation tests, observed values evaluation of optimal model parameters and predicted categories, bootstrap values evaluation of optimal model parameters and predicted cross-validated categories. & \href{https://cran.r-project.org/package=packMBPLSDA}{packMBPLSDA} & CRAN\\
Robust identification of time intervals are significantly different between groups. & \href{NA}{OmicsLonDA} & BioC\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Omics Data integration}}\\
Multiple co-inertia analysis of omics datasets (MCIA) is a multivariate approach for visualization and integration of multi-omics datasets. The MCIA method is not dependent on feature annotation therefore can extract important features even when there are not present across all datasets. & \href{http://bioconductor.org/packages/release/bioc/html/omicade4.html}{omicade4} & BioC\\
STATegRa combines information in multiple omics datasets to evaluate the reproducibility among samples and across experimental condition using component analysis (omicsNPC implements the NonParametric Combination) and clustering. & \href{NA}{STATegRa} & BioC\\
\rowcolor{gray!6}  Statistical framework supporting many different types of multivariate analyses (e.g. PCA, PLS, CCA, PLS-DA, etc.). & \href{https://cran.r-project.org/web/packages/mixOmics/index.html}{mixOmics} & CRAN\\
STatistics in R Using Class Templates - Classes for building statistical workflows using methods, models and validation objects. Provides mechanism for STATO integration. & \href{http://www.github.com/computational-metabolomics/struct}{STRUCT} & GitHub\\
\rowcolor{gray!6}  Multi-omics base clases integrable with commonly used R Bioconductor objects for omics data; container that holds omics results. & \href{NA}{MultiDataSet} & BioC\\
Identifies analyte-analyte (e.g. gene-metabolite) pairs whose relationship differs by phenotype (e.g. positive correlation in one phenotype, negative or no correlation in another). The software is also accessible as a user-friendly interface at intlim.bmi.osumc.edu. & \href{https://github.com/mathelab/IntLIM}{IntLIM} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Missing value imputation}}\\
Mixture-model for accounting for data missingness'. & \href{NA}{metabomxtr} & BioC\\
Kernel-Based Metabolite Differential Analysis provides a kernel-based score test to cluster metabolites between treatment groups, in order to handle missing values. & \href{NA}{KMDA} & CRAN\\
\rowcolor{gray!6}  Visualization and imputation of missing values. VIM provides methods for the evaluation and visualization of the type and patterns of missing data. The included imputation approaches are kNN, Hot-Deck, iterative robust model-based imputation (IRMI), fast matching/imputation based on categorical variables and regression imputation. & \href{NA}{VIM} & CRAN\\
Graphical user interface for VIM. & \href{NA}{VIMGUI} & CRAN\\
\rowcolor{gray!6}  kNN based imputation for microarray data. & \href{NA}{impute} & BioC\\
Bootstrap based algorithm and diagnostics for fast and robust multiple imputation for cross sectional, time series or combined cross sectional and time series data. & \href{NA}{Amelia} & CRAN\\
\rowcolor{gray!6}  Algorithms and diagnostics for the univariate imputation of time series data. & \href{NA}{imputeTS} & CRAN\\
Methods for the Imputation of incomplete continuous or categorical datasets. missMDA allows missing data imputation using in categorial, continuous or mixed-type datasets using PCA, CA, a multiple correspondence analysis (MCA) model, a multiple factor analysis (MFA) model or factorial analysis for mixed data (FAMD). & \href{NA}{missMDA} & CRAN\\
\rowcolor{gray!6}  Random forest based missing data imputation for mixed-type, nonparametric data. An out-of-bag (OOB) error estimate is used for model optimization. & \href{NA}{missForest} & CRAN\\
Multivariate imputation by chained equations using fully conditional specifications (FCS) for categorical, continuous and binary datasets. It includes various diagnostic plots for the evaluation of the imputation quality. & \href{NA}{mice} & CRAN\\
\rowcolor{gray!6}  Missing data imputation using an approximate Bayesian framework. Diagnostic algorithms are included to analyze the models, the assumptions of the imputation algorithm and the multiply imputed datasets. & \href{NA}{mi} & CRAN\\
Iterative Gibbs sampler based left-censored missing value imputation. & \href{NA}{GSimp} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Multiple workflow steps}}\\
Missing value imputation, filtering, normalisation and averaging of technical replications. & \href{https://sourceforge.net/projects/msprep/}{MSPrep} & SF\\
HCA, Fold change analysis, heat maps, linear models (ordinary and empirical Bayes), PCA and volcano plots. Also log transformation, missing value replacement and methods for normalisation. Cross-contribution compensating multiple internal standard normalisation (ccmn) and remove unwanted variation (ruv2). & \href{https://cran.r-project.org/package=metabolomics}{metabolomics} & CRAN\\
\rowcolor{gray!6}  Data processing, normalization, statistical analysis, metabolite set enrichment analysis, metabolic pathway analysis, and biomarker analysis. & \href{https://github.com/xia-lab/MetaboAnalystR}{MetaboAnalystR} & GitHub\\
Pipeline for metabolomics data pre-processing, with particular focus on data representation using univariate and multivariate statistics. Built on already published functions. & \href{https://cran.r-project.org/web/packages/muma/index.html}{muma} & GitHub\\
\rowcolor{gray!6}  Framework for multiomics experiments. Identifies sources of variability in the experiment and performs additional analysis (identification of subgroups, data imputation, outlier detection). & \href{https://www.bioconductor.org/packages/release/bioc/html/MOFA.html}{MOFA} & BioC\\
Performs entry-level differential analysis on metabolomics data. & \href{https://github.com/andreasmock/MetaboDiff}{MetaboDiff} & GitHub\\
\rowcolor{gray!6}  STRUCT wrappers for filtering, normalisation, missing value imputation, glog transform, HCA, PCA, PLSDA, PLSR, t-test, fold-change, ANOVA, Mixed Effects, post-hoc tests & \href{https://github.com/computational-metabolomics/structToolbox}{STRUCTToolbox} & GitHub\\
Data transformation, filtering of feature and/or samples and data normalization. Quality control processing, statistical analysis and visualization of MS data. & \href{https://github.com/pmartR/pmartR}{pmartR} & GitHub\\
\rowcolor{gray!6}  Quality control, signal drift and batch correction, transformation, univariate hypothesis testing. & \href{https://github.com/fgiacomoni/metabolis}{metabolis} & GitHub\\
Missing value filtering and imputation, zero value filtering, data normalization, data integration, data quality assessment, univariate statistical analysis, multivariate statistical analysis such as PCA and PLS-D and potential marker selection & \href{https://github.com/MetabLAB/MetCleaning}{MetCleaning} & GitHub\\
\rowcolor{gray!6}  Univariate analysis (linear model), PCA, clustered heatmap, and partial correlation network analysis. Based on classes from the Metabase package(Zhu 2019). & \href{https://github.com/zhuchcn/ShinyMetabase}{ShinyMetabase} & GitHub\\
Outlier detection, PCA, drift correction,visualization, missing value imputation,classification. & \href{https://cran.r-project.org/package=MetabolomicsBasics}{MetabolomicsBasics} & CRAN\\
\rowcolor{gray!6}  Pre-processing, differential compound identification and grouping, traditional PK parameters calculation, multivariate statistical analysis, correlations, cluster analyses and resulting visualization. & \href{https://cran.r-project.org/package=polyPK}{polyPK} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{handling-of-molecule-structures-and-chemical-structure-databases}{%
\subsection{Handling of molecule structures and chemical structure databases}\label{handling-of-molecule-structures-and-chemical-structure-databases}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab7}R Packages for molecule structures and chemical structure databases.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab7}R Packages for molecule structures and chemical structure databases. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Structure representation and manipulation}}\\
Subset of functions from the Chemistry Development Kit. Provide a computer readable representation of molecular structures and provide functions to import structures from different molecule structure description formats, manipulate structures, visualize structures and calculate properties and molecular fingerprints. & \href{https://cran.r-project.org/package=rcdk}{rcdk} & CRAN\\
Similar torcdkin functionality and provides more fingerprints and clustering methods and provides additional tools through querying the ChemMine Tools web service. & \href{https://doi.org/doi:10.18129/B9.bioc.ChemmineR}{ChemmineR} & BioC\\
\rowcolor{gray!6}  Provides conversion of structure representation through OpenBabel. & \href{https://doi.org/doi:10.18129/B9.bioc.ChemmineOB}{ChemmineOB} & BioC\\
Exposes functionalities of the RDKit library, including reading and writing of SF files and calculating a few physicochemical properties. & \href{https://github.com/pauca/RRDKit}{RRDKit} & GitHub\\
\rowcolor{gray!6}  Read and write InChI and InChIKey from and torcdk. & \href{https://github.com/rajarshi/cdkr}{rinchi} & GitHub\\
Maximum Common Substructure Searching using ChemmineR structures. & \href{https://doi.org/doi:10.18129/B9.bioc.fmcsR}{FmcsR} & BioC\\
\rowcolor{gray!6}  Basic cheminformatics functions tailored for mass spectrometry applications, enhancing functionality available in other packages likercdk, enviPat, RMassBank etc. & \href{https://github.com/schymane/RChemMass}{RChemMass} & GitHub\\
Provides fingerprinting methods forrcdk. & \href{https://cran.r-project.org/package=fingerprint}{fingerprint} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Database queries}}\\
Calculation of molecular properties. & \href{https://github.com/cambDI/camb}{camb} & GitHub\\
Querying information from PubChem. & \href{https://cran.r-project.org/package=rpubchem}{Rpubchem} & CRAN\\
\rowcolor{gray!6}  Querying information from various web services (CACTUS, CTS, PubChem, ChemSpider) as part of compound list generation. & \href{https://doi.org/doi:10.18129/B9.bioc.RMassBank}{RMassBank} & BioC\\
Querying information from a large number of databases. & \href{https://cran.r-project.org/package=webchem}{webchem} & CRAN\\
\rowcolor{gray!6}  R Interface to the ClassyFire REST API. & \href{https://cran.r-project.org/web/packages/classyfireR/index.html}{classyfireR} & CRAN\\
Allows mapping of identifiers from one database to another, for metabolites, genes, proteins, and interactions. & \href{https://doi.org/10.18129/B9.bioc.BridgeDbR}{BridgeDbR} & BioC\\
\rowcolor{gray!6}  Define utilities for exploration of human metabolome database, including functions to retrieve specific metabolite entries and data snapshots with pairwise associations. & \href{https://doi.org/doi:10.18129/B9.bioc.hmdbQuery}{hmdbQuery} & BioC\\
Parsers for many compound databases including HMDB, MetaCyc, ChEBI, FooDB, Wikidata, WikiPathways, RIKEN respect, MaConDa, T3DB, KEGG, Drugbank, LipidMaps, MetaboLights, Phenol-Explorer, MassBank. & \href{https://github.com/UMCUGenetics/MetaDBparse}{MetaDBparse} & GitHub\\
\rowcolor{gray!6}  Functionality to create and use compound databases generated from (mostly publicly) available resources such as HMDB, ChEBI and PubChem. & \href{https://github.com/EuracBiomedicalResearch/CompoundDb}{CompoundDb} & GitHub\\
Standardized and extensible framework to query chemical and biological databases. & \href{https://github.com/pkrog/biodb}{biodb} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{network-analysis-and-biochemical-pathways}{%
\subsection{Network analysis and biochemical pathways}\label{network-analysis-and-biochemical-pathways}}

\hypertarget{network-infrastructure-and-analysis}{%
\subsubsection{Network infrastructure and analysis}\label{network-infrastructure-and-analysis}}

\hypertarget{metabolite-annotation}{%
\subsubsection{Metabolite annotation}\label{metabolite-annotation}}

\hypertarget{generation-of-metabolic-networks}{%
\subsubsection{Generation of metabolic networks}\label{generation-of-metabolic-networks}}

\hypertarget{pathway-analysis}{%
\subsubsection{Pathway analysis}\label{pathway-analysis}}

\hypertarget{pathway-resources-and-interfaces}{%
\subsubsection{Pathway resources and interfaces}\label{pathway-resources-and-interfaces}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab8}R packages for network analysis and Biochemical pathways.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab8}R packages for network analysis and Biochemical pathways. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Network infrastructure and analysis}}\\
Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=igraph}{igraph} & CRAN\\
Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=tidygraph}{tidygraph} & CRAN\\
\rowcolor{gray!6}  Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=statnet}{statnet} & CRAN\\
Interactive visualization and manipulation of networks. & \href{https://bioconductor.org/packages/release/bioc/html/RedeR.html}{RedeR} & BioC\\
\rowcolor{gray!6}  Comparison of correlation networks from two experiments. & \href{https://cran.r-project.org/package=DiffCorr}{DiffCorr} & CRAN\\
Correlation-based networks from metabolomics data and analysis tools. & \href{https://bioconductor.org/packages/release/bioc/html/BioNetStat.html}{BioNetStat} & BioC\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Annotation}}\\
Putative annotation of unknowns in MS1 data. & \href{http://bioconductor.org/packages/release/bioc/html/MetNet.html}{MetNet} & BioC\\
Putative annotation of unknowns in MS1 data. & \href{https://sourceforge.net/projects/xmsannotator/}{xMSAnnotator} & SF\\
\rowcolor{gray!6}  Putative annotation of unknowns using MS1 and MS2 data. & \href{https://github.com/ZhuMSLab/MetDNA}{MetDNA} & GitHub\\
Visualization of spectral similarity networks, putative annotation of unknowns using MS2 data. & \href{https://bioconductor.org/packages/release/bioc/html/MetCirc.html}{MetCirc} & BioC\\
\rowcolor{gray!6}  Putative annotation of unknowns using MS2 data, clustering of MS2 data. & \href{https://bioconductor.org/packages/devel/bioc/html/CluMSID.html}{CluMSID} & BioC\\
Putative annotation of unknowns using MS2 data. & \href{https://github.com/WMBEdmands/compMS2Miner}{compMS2Miner} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Generation of metabolite networks}}\\
Biochemical reaction networks, spectral and structural similarity networks. & \href{http://dgrapov.github.io/MetaMapR/}{MetaMapR} & GitHub\\
Correlation-based networks, structural similarity networks. & \href{http://kwanjeeraw.github.io/metabox/}{Metabox} & GitHub\\
\rowcolor{gray!6}  Targeted metabolome-wide association studies. & \href{https://sourceforge.net/projects/metabnet/}{MetabNet} & SF\\
Generation of scale-free correlation-based networks. & \href{https://cran.r-project.org/package=WGCNA}{WGCNA} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Pathway analysis}}\\
Analysis of -omics data, pathway, transcription factor and target gene identification. & \href{NA}{pwOmics} & BioC\\
MSEA a metabolite set enrichment analysis with factor loading in principal component analysis. & \href{https://cran.r-project.org/package=mseapca}{mseapca} & CRAN\\
\rowcolor{gray!6}  Enrichment analysis of a list of affected metabolites. & \href{https://cran.r-project.org/package=tmod}{tmod} & CRAN\\
Network-based enrichment analysis of a list of affected metabolites. & \href{https://bioconductor.org/packages/release/bioc/html/FELLA.html}{FELLA} & BioC\\
\rowcolor{gray!6}  Pathway-based enrichment analysis of a list of affected metabolites. & \href{https://cran.r-project.org/package=CePa}{CePa} & CRAN\\
Differential analysis, modules/sub-pathway identification using networks. & \href{https://github.com/andreasmock/MetaboDiff}{MetaboDiff} & GitHub\\
\rowcolor{gray!6}  Integrates metabolic networks and RNA-seq data to construct condition-specific series of metabolic sub-networks and applies to gene set enrichment analysis & \href{https://cran.r-project.org/package=metaboGSE}{metaboGSE} & CRAN\\
Differential analysis. & \href{https://bioconductor.org/packages/release/bioc/html/SDAMS.html}{SDAMS} & BioC\\
\rowcolor{gray!6}  Biomarker identification. & \href{https://cran.r-project.org/package=lilikoi}{lilikoi} & CRAN\\
Biomarker identification. & \href{https://bioconductor.org/packages/release/bioc/html/INDEED.html}{INDEED} & BioC\\
\rowcolor{gray!6}  Biomarker identification. & \href{https://github.com/krumsieklab/MoDentify}{MoDentify} & GitHub\\
Pathway activity profiling. & \href{NA}{PAPi} & BioC\\
\rowcolor{gray!6}  Pathway activity profiling. & \href{https://www.bioconductor.org/packages/devel/bioc/html/pathwayPCA.html}{pathwayPCA} & BioC\\
Flux balance analysis. & \href{https://bioconductor.org/packages/release/bioc/html/BiGGR.html}{BiGGR} & BioC\\
\rowcolor{gray!6}  Flux balance analysis. & \href{https://cran.r-project.org/package=abcdeFBA}{abcdeFBA} & CRAN\\
Flux balance analysis. & \href{https://cran.r-project.org/package=sybil}{sybil} & CRAN\\
\rowcolor{gray!6}  Flux balance analysis. & \href{https://cran.r-project.org/package=fbar}{fbar} & CRAN\\
Identification of affected pathway from phenotype data (interface with graphite). & \href{https://www.bioconductor.org/packages/release/bioc/html/SPIA.html}{SPIA} & BioC\\
\rowcolor{gray!6}  Identification of affected pathway from phenotype data (interface with graphite). & \href{https://www.bioconductor.org/packages/release/bioc/html/clipper.html}{clipper} & BioC\\
Interface to PathVisio and WikiPathways and pathway analysis and enrichment. & \href{https://github.com/PathVisio/RpathVisio}{RPathVisio} & GitHub\\
\rowcolor{gray!6}  Enrichment analysis of a list of genes and metabolites. & \href{https://github.com/Mathelab/RaMP-DB}{RaMP} & GitHub\\
Simulation of longitudinal metabolomics data based on an underlying biological network & \href{https://cran.r-project.org/package=MetaboLouise}{MetaboLouise} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Pathway resources and interfaces}}\\
BioPax parser and representation in R. & \href{https://www.bioconductor.org/packages/release/bioc/html/rBiopaxParser.html}{rBiopaxParser} & BioC\\
Interface to KEGG, Biocarta, Reactome, NCI/Nature Pathway Interaction Database, HumanCyc, Panther, SMPDB and PharmGKB. & \href{http://bioconductor.org/packages/release/bioc/html/graphite.html}{graphite} & BioC\\
\rowcolor{gray!6}  Interface to NCI Pathways Database. & \href{https://www.bioconductor.org/packages/release/bioc/html/NCIgraph.html}{NCIgraph} & BioC\\
Interface to KEGG. & \href{https://bioconductor.org/packages/release/bioc/html/pathview.html}{pathview} & BioC\\
\rowcolor{gray!6}  Interface to KEGG. & \href{https://www.bioconductor.org/packages/release/bioc/html/KEGGgraph.html}{KEGGgraph} & BioC\\
Interface to systems biology markup language (SBML). & \href{https://www.bioconductor.org/packages/release/bioc/html/SBMLR.html}{SBMLR} & BioC\\
\rowcolor{gray!6}  Interface to systems biology markup language (SBML). & \href{https://bioconductor.org/packages/release/bioc/html/rsbml.html}{rsbml} & BioC\\
Interface to Gaggle-enabled software (Cytoscape, Firegoose, Gaggle Genome browser). & \href{https://bioconductor.org/packages/release/bioc/html/gaggle.html}{gaggle} & BioC\\
\rowcolor{gray!6}  Interface to molecular interaction databases. & \href{https://www.bioconductor.org/packages/release/bioc/html/PSICQUIC.html}{PSICQUIC} & BioC\\
Interface to KEGG REST server. & \href{http://bioconductor.org/packages/release/bioc/html/KEGGREST.html}{KEGGREST} & BioC\\
\rowcolor{gray!6}  Interface to BioPAX OWL files and the Pathway Commons (PW) molecular interaction database. & \href{http://bioconductor.org/packages/release/bioc/html/paxtoolsr.html}{paxtoolsr} & BioC\\
Interface to WikiPathways. & \href{https://bioconductor.org/packages/release/bioc/html/rWikiPathways.html}{rWikiPathways} & BioC\\
\rowcolor{gray!6}  Database that integrates metabolite and gene biological pathways from HMDB, KEGG, Reactome, and WikiPathways. Includes user-friendly R Shiny web application for queries and pathway enrichment analysis. & \href{https://github.com/Mathelab/RaMP-DB/}{RaMP-DB} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{multifunctional-workflows}{%
\subsection{Multifunctional workflows}\label{multifunctional-workflows}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab9}R packages with multifunctional workflows.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab9}R packages with multifunctional workflows. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  Convenience wrapper for pre-processing tools (XCMS, CAMERA) and a number of statistical analyses. & \href{https://www.bioconductor.org/packages/release/bioc/html/MAIT.html}{MAIT} & BioC\\
Preprocessing (XCMS), replicate merging, noise, blank and missingness filtering, feature grouping, annotation of known compounds, isotopic labeling analysis, annotation from KEGG or HMDB, common biotransformations, probabilistic putative metabolite annotation. & \href{http://mzmatch.sourceforge.net/}{mzMatch} & SF\\
\rowcolor{gray!6}  XCMS and CAMERA based workflow for non-targeted processing of LC-MS datasets, It includes pre-processing, peak picking,peak filtering, data normalization and descriptive statistics calculation. & \href{https://github.com/search?q=MStractor}{MStractor} & GitHub\\
Performs simultaneous raw data to mzXML conversion (MSConvert), peak-picking, automatic PCA outlier detection and statistical analysis, visualization and possible MS2 target list determination during an MS1 metabolomic profiling experiment. & \href{https://github.com/WMBEdmands/simExTargId}{simExTargId} & GitHub\\
\rowcolor{gray!6}  Pre-processing of large LC-MS datasets. Performs automatic PCA with iterative automatic outlier removal and, clustering analysis and biomarker discovery. & \href{https://github.com/WMBEdmands/MetMSLine}{MetMSLine} & GitHub\\
\addlinespace
Workflow for the systematic analysis of 1H NMR metabolomics dataset in quantitative genetics. Performs pre-processing, mQTL mapping, metabolites structural assignment and offers data visualisation tools. & \href{https://www.bioconductor.org/packages/3.5/bioc/html/mQTL.NMR.html}{mQTL.NMR} & BioC\\
\rowcolor{gray!6}  Workflow for pre-processing, qc, annotation and statistical data analysis of LC-MS and GC-MS based metabolomics data to be submitted to public repositories. & \href{https://github.com/rmylonas/MetaDB}{MetaDB} & GitHub\\
Specmine is a framework mainly built on a number of already published packages. It supports data processing form different analytical platforms (LC-MS, GC-MS, NMR, IR, UV-Vis). & \href{https://github.com/cran/specmine}{specmine} & GitHub\\
\rowcolor{gray!6}  Common interface for a number of different MS based data processing software. It covers various aspects, such as data preparation and data extraction, formula calculation, compound identification and reporting. & \href{https://github.com/rickhelmus/patRoon}{patRoon} & GitHub\\
Processing of high resolution of LC-MS data for environmental trend analysis. & \href{https://zenodo.org/record/1213098}{enviMass} & Zenodo\\
\addlinespace
\rowcolor{gray!6}  Workflow for preprocessing of LC-HRMS data, suspect screening, screening for transformation products using combinatorial prediction, and interactive filtering based on ratios between sample groups. & \href{https://github.com/meowcat/RMassScreening}{RMassScreening} & GitHub\\
Workflow to perform pre-processing, statistical analysis and metabolite identifications based on database search of detected spectra. & \href{https://github.com/tohweizhong/MetaboNexus}{MetaboNexus} & GitHub\\
\rowcolor{gray!6}  Shiny-based platform to extract differential features from LC-MS data, includes XCMS-based feature detection, statistical analysis, prediction of molecular formulas, annotation of MS2 spectra, MS2 molecular networking and chemical compound database search. & \href{https://github.com/mjhelf/METABOseek}{METABOseek} & GitHub\\
RShiny interface to Metabolomics packages {\&} MetaboAnalyst scripts. & \href{https://github.com/UMCUGenetics/MetaboShiny/}{MetaboShiny} & GitHub\\
\rowcolor{gray!6}  Preprocessing and visualizing for LC-MS data, as well as statistical analyses, mainly based on univariate linear models. & \href{https://github.com/antonvsdata/amp}{amp} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{user-interfaces-and-workflow-management-systems}{%
\subsection{User interfaces and workflow management systems}\label{user-interfaces-and-workflow-management-systems}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab11}Packages to interface R with other languages and workflow environments}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab11}Packages to interface R with other languages and workflow environments \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  Given an R function and its manual page, make the documented function available in Galaxy. & \href{http://bioconductor.org/packages/release/bioc/html/RGalaxy.html}{RGalaxy} & BioC\\
Integration of R and C++. Many R data types and objects can be mapped back and forth to C++ equivalents. & \href{https://cran.r-project.org/package=Rcpp}{Rcpp} & CRAN\\
\rowcolor{gray!6}  Low-Level R to Java Interface. & \href{https://cran.r-project.org/package=rJava}{rJava} & CRAN\\
Interface to 'Python' modules, classes, and functions and translation between R and Python objects. & \href{https://cran.r-project.org/package=reticulate}{reticulate} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{metabolomics-data-sets}{%
\subsection{Metabolomics data sets}\label{metabolomics-data-sets}}

\begin{longtable}{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab12}Metabolomics data sets packaged as R packages.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab12}Metabolomics data sets packaged as R packages. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{LC-MS}}\\
12 HPLC-MS NetCDF files (Agilent 1100 LC-MSD SL). & \href{NA}{faahKO} & BioC\\
16 UPLC-MS mzData files (Bruker microTOFq). & \href{NA}{mtbls2} & BioC\\
\rowcolor{gray!6}  12 UPLC-MS mzML files (AB Sciex TripleTOF 5600, SWATH mode). & \href{NA}{mtbls297} & GitHub\\
Different raw MS files (LTQ, TripleQ, FTICR, Orbitrap, QTOF) some in different formats (mzML, mzXML, mzData, mzData.gz, NetCDF, mz5). Also mzid format from proteomics. & \href{NA}{msdata} & BioC\\
\rowcolor{gray!6}  Metadata and DDA MS/MS spectra of 15 narcotics standards (LTQ Orbitrap XL). & \href{https://bioconductor.org/packages/release/data/experiment/html/RMassBankData.html}{RMassBankData} & BioC\\
183 x 109 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/ropls.html}{ropls} & BioC\\
\rowcolor{gray!6}  69 x 5,501 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/biosigner.html}{biosigner} & BioC\\
40 x 1,632 peak table. & \href{https://cran.r-project.org/web/packages/BioMark/index.html}{BioMark} & CRAN\\
\rowcolor{gray!6}  Raw MS files from a set of blanks and standards that contain common environmental contaminants (acquired with Bruker maXis 4G). & \href{https://github.com/rickhelmus/patRoonData}{patRoonData} & GitHub\\
Proteomics, metabolomics GC-MS and Lipidomics data from Calu-3 cell culture; 3 mockulum treated and 9 MERS-CoV treated; Time point, 18 hour from MassIVE dataset ids MSV000079152, MSV000079153, MSV000079154. & \href{https://github.com/pmartR/pmartRdata}{pmartRdata} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{FIA-MS}}\\
6 mzML files (human plasma spiked with 40 compounds acquired in positive mode on an orbitrap fusion). & \href{NA}{plasFIA} & BioC\\
mzML files (Thermo Exactive) from comparison of leaf tissue from 4 B. distachyon ecotypes with Flow-infusion electrospray ionisation-high resolution mass spectrometry (FIE-HRMS). Also includes data sets with 10 technical injections of human urine and another 10 injections from leaf tissue (ecotype ABR1). & \href{NA}{metaboData} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{GC-MS}}\\
52 x 154 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/pcaMethods.html}{pcaMethods} & BioC\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{NMR}}\\
18 x 189 peak table. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
\rowcolor{gray!6}  33 x 164 peak table. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
ASICSdata: 1D NMR spectra for ASICS. & \href{https://www.bioconductor.org/packages/release/data/experiment/vignettes/ASICSdata/inst/doc/ASICSdata.html}{ASICSdata} & BioC\\*
\end{longtable}

\pagebreak

\hypertarget{conclusions}{%
\section{Conclusions}\label{conclusions}}

\newpage

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-emwas_2019}{}%
1. Emwas, A.-H.; Roy, R.; McKay, R.T.; Tenori, L.; Saccenti, E.; Gowda, G.A.N.; Raftery, D.; Alahmari, F.; Jaremko, L.; Jaremko, M. et al. NMR spectroscopy for metabolomics research. \emph{Metabolites} \textbf{2019}, \emph{9}.

\leavevmode\hypertarget{ref-lammerhofer_2013}{}%
2. \emph{Metabolomics in practice: Successful strategies to generate and analyze metabolic data}; Lämmerhofer, M., Weckwerth, W., Eds.; Wiley-VCH Verlag GmbH \& Co. KGaA: Weinheim, Germany, 2013; ISBN \href{https://worldcat.org/isbn/9783527330898}{9783527330898}.

\leavevmode\hypertarget{ref-villasboas_2007}{}%
3. Villas-Boas, S.G.; Nielsen, J.; Smedsgaard, J.; Hansen, M.A.E.; Roessner-Tunali, U. \emph{Metabolome analysis: An introduction}; 1st ed.; Wiley, John \& Sons, Incorporated, 2007; p. 319; ISBN \href{https://worldcat.org/isbn/978-0-471-74344-6}{978-0-471-74344-6}.

\leavevmode\hypertarget{ref-wehrens_2019}{}%
4. \emph{Metabolomics: Practical guide to design and analysis}; Wehrens, R., Salek, R., Eds.; Chapman \& hall/CRC mathematical and computational biology; Chapman; Hall/CRC, 2019; ISBN \href{https://worldcat.org/isbn/1498725260}{1498725260}.

\leavevmode\hypertarget{ref-InternationalMetabolomicsSociety_2019}{}%
5. International Metabolomics Society Free Tools \& Learning Resources - Metabolomics Society Wiki.

\leavevmode\hypertarget{ref-salek_website_nd}{}%
6. Salek, R.; Emery, L.; Beisken, S. Metabolomics: An introduction EMBL-EBI train online.

\leavevmode\hypertarget{ref-rcoredevelopmentteam_website_2018}{}%
7. R Core Development Team R: A language and environment for statistical computing 2018.

\leavevmode\hypertarget{ref-spicer_website_2018}{}%
8. Spicer, R. GitHub - RASpicer/MetabolomicsTools 2018.

\leavevmode\hypertarget{ref-spicer_2017}{}%
9. Spicer, R.; Salek, R.M.; Moreno, P.; Cañueto, D.; Steinbeck, C. Navigating freely-available software tools for metabolomics analysis. \emph{Metabolomics : Official journal of the Metabolomic Society} \textbf{2017}, \emph{13}, 106.

\leavevmode\hypertarget{ref-misra_2016}{}%
10. Misra, B.B.; Hooft, J.J.J. van der Updates in metabolomics tools and resources: 2014-2015. \emph{Electrophoresis} \textbf{2016}, \emph{37}, 86--110.

\leavevmode\hypertarget{ref-misra_2017}{}%
11. Misra, B.B.; Fahrmann, J.F.; Grapov, D. Review of emerging metabolomic tools and resources: 2015-2016. \emph{Electrophoresis} \textbf{2017}, \emph{38}, 2257--2274.

\leavevmode\hypertarget{ref-misra_2018}{}%
12. Misra, B.B. New tools and resources in metabolomics: 2016-2017. \emph{Electrophoresis} \textbf{2018}, \emph{39}, 909--923.

\leavevmode\hypertarget{ref-misra_website_2017}{}%
13. Misra, B. GitHub - biswapriyamisra/metabolomics: Tools databases resources in metabolomics \& integrated omics in 2015-2016 2017.

\leavevmode\hypertarget{ref-kannan_2016}{}%
14. Kannan, L.; Ramos, M.; Re, A.; El-Hachem, N.; Safikhani, Z.; Gendoo, D.M.A.; Davis, S.; Gomez-Cabrero, D.; Castelo, R.; Hansen, K.D. et al. Public data and open source tools for multi-assay genomic investigation of disease. \emph{Briefings in Bioinformatics} \textbf{2016}, \emph{17}, 603--615.

\leavevmode\hypertarget{ref-blaenovi_2018}{}%
15. Blaženović, I.; Kind, T.; Ji, J.; Fiehn, O. Software tools and approaches for compound identification of LC-MS/MS data in metabolomics. \emph{Metabolites} \textbf{2018}, \emph{8}.

\leavevmode\hypertarget{ref-mullen_website_2019}{}%
16. Mullen, K. CRAN task view: Chemometrics and computational physics 2019.

\leavevmode\hypertarget{ref-gentleman_2004}{}%
17. Gentleman, R.C.; Carey, V.J.; Bates, D.M.; Bolstad, B.; Dettling, M.; Dudoit, S.; Ellis, B.; Gautier, L.; Ge, Y.; Gentry, J. et al. Bioconductor: Open software development for computational biology and bioinformatics. \emph{Genome Biology} \textbf{2004}, \emph{5}, R80.

\leavevmode\hypertarget{ref-bioconductor_website_nda}{}%
18. Bioconductor Bioconductor - BiocViews.

\leavevmode\hypertarget{ref-thecomprehensiverarchivenetwork_website_nd}{}%
19. The Comprehensive R Archive Network CRAN repository policy.

\leavevmode\hypertarget{ref-bioconductor_website_nd}{}%
20. Bioconductor Bioconductor - developers.

\leavevmode\hypertarget{ref-theul_2009}{}%
21. Theußl, S.; Zeileis, A. Collaborative software development using r-forge. \emph{The R journal} \textbf{2009}, \emph{1}, 9.

\leavevmode\hypertarget{ref-boettiger_2015}{}%
22. Boettiger, C.; Chamberlain, S.; Hart, E.; Ram, K. Building software, building community: Lessons from the rOpenSci project. \emph{Journal of open research software} \textbf{2015}, \emph{3}.

\leavevmode\hypertarget{ref-devries_website_2015b}{}%
23. Vries, A. de; Rickert, J. The network structure of r packages on CRAN \& BioConductor 2015.

\leavevmode\hypertarget{ref-devries_website_2015a}{}%
24. Vries, A. de Differences in the network structure of CRAN and BioConductor (revolutions) 2015.

\leavevmode\hypertarget{ref-devries_website_2015}{}%
25. Vries, A. de GitHub - andrie/cran-network-structure: Scripts used for my UseR!2015 presentation on the network structure of CRAN 2015.

\leavevmode\hypertarget{ref-neumann_website_2019}{}%
26. Neumann, S. GitHub - sneumann/metaRbolomics: Metabolomics in r and bioconductor 2019.

\newpage

\hypertarget{appendices}{%
\section{Appendices}\label{appendices}}

\newpage

\hypertarget{appendix-1-the-msp-file-format-and-package-support}{%
\subsection*{Appendix 1: The MSP File Format and package support}\label{appendix-1-the-msp-file-format-and-package-support}}
\addcontentsline{toc}{subsection}{Appendix 1: The MSP File Format and package support}

\begin{longtable}[]{@{}l@{}}
\toprule
\endhead
\begin{minipage}[t]{0.38\columnwidth}\raggedright
Name: unknown\\
Num Peaks: 2\\
85.345 100; 76.321 50;\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\textbf{Listing S1:} Example for the basic NIST format.

\leavevmode\newline\leavevmode\newline\leavevmode\newline\leavevmode\newline

\begin{tabular}{>{\raggedright\arraybackslash}p{22em}|>{\raggedright\arraybackslash}p{22em}}
\toprule
Name: 1-Methylhistidine\newline Synon: (2S)-2-amino-3-(1-methyl-1H-imidazol-4-yl)propanoic acid\newline SYNON: {\$}:00in-source\newline DB{\#}: HMDB0000001{\_{}}c{\_{}}ms{\_{}}1469\newline InChIKey: BRMWTNUJHUMWMS-LURJTMIESA-N\newline Instrument{\_{}}type: GC-MS\newline Retention{\_{}}index: 1807.71\newline Formula: C7H11N3O2\newline MW: 169\newline ExactMass: 169.0851\newline Comments: "column=5{\%}-phenyl-95{\%}-dimethylpolysiloxane capillary column" "derivatization type=2 TMS" "derivatization formula=C13H27N3O2Si2" "derivative mw=313.544" "retention index=1807.71" "retention index type=based on 9 n-alkanes (C10–C36)" "instrument type=GC-MS" "chromatography type=GC" "cas number=332-80-9" "molecular formula=C7H11N3O2" "total exact mass=169.085126592" "InChIKey=BRMWTNUJHUMWMS-LURJTMIESA-N"\newline Num Peaks: 10\newline  70  0.014;  71  0.007;  72  0.02;  76  0.008;  77  0.008;\newline  78  0.002;  79  0.003;  80  0.005;  81  0.108;  82  0.017; & NAME: Aspartame; LC-ESI-ITFT; MS2; CE\newline PRECURSORMZ: 295.128848\newline PRECURSORTYPE: [M+H]+\newline INSTRUMENTTYPE: LC-ESI-ITFT\newline SMILES: COC(=O)C(CC1=CC=CC=C1)N=C(O)C(N)CC(O)=O\newline INCHIKEY: IAOZJIPTCAWIRG-UHFFFAOYNA-N\newline Ontology: Peptides\newline COLLISIONENERGY: 35\newline FORMULA: C14H18N2O5\newline RETENTIONTIME:  \newline IONMODE: Positive\newline Comment: registered in MassBank\newline Num Peaks: 9\newline 120.0804    13\newline 180.10201    138\newline 217.0968    14\newline 235.10789    390\newline 245.0921    274\newline 260.09171    132\newline 263.1026    286\newline 277.11859    1000\newline 278.1022    28\\
\hline
\rowcolor{white}  {\bf Listing S2:} Example for the canonical NIST format. & {\bf Listing S3:} RIKEN PRIME msp format example.\\
\bottomrule
\end{tabular}

\leavevmode\newline\leavevmode\newline\leavevmode\newline\leavevmode\newline
\newpage

\textbf{Table S1:} Overview of MS/MS handling in different R packages. `-' means not available, for the remaining entries see the text above.

\begin{longtable}[]{@{}llll@{}}
\toprule
\begin{minipage}[b]{0.09\columnwidth}\raggedright
package\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright
read msp\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright
write msp\strut
\end{minipage} & \begin{minipage}[b]{0.46\columnwidth}\raggedright
spectral matching and additional information\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.09\columnwidth}\raggedright
baitmet\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; cosine, Stein \& Scott composite similarity product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
compMS2Miner\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST, RIKEN PRIME msp\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
RIKEN PRIME msp\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
enviGCMS\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
erah\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; cosine\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
flagme\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
metaMS\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST; slow\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs DB, N vs DB; proprietary\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MatchWeiz\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; X-Rank\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MetCirc\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs N; normalized dot product;\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
will switch to MSnbase functions soon\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MSeasy\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; Queries the NIST mass spectral search tool\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MSnbase\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
**\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
**\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs 1, N vs N; dot product and more, user def.\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
msPurity\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
OrgMassSpecR\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs 1; normalized dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
RAMClustR\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
RAMClustR can import and utilize spectrum similarities from MS-FINDER;\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
rTANDEM\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product; R wrapper for X!Tandem software\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
SwathXtend\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
- (PeakView / OpenSWATH)\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
- (PeakView / OpenSWATH)\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
TargetSearch\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST (with error)\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; RI-based\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\newpage

\hypertarget{appendix-2-metarbolomics-dependencies-network}{%
\subsection*{Appendix 2: metaRbolomics dependencies network}\label{appendix-2-metarbolomics-dependencies-network}}
\addcontentsline{toc}{subsection}{Appendix 2: metaRbolomics dependencies network}

\hypertarget{libraries-and-settings}{%
\subsubsection*{Libraries and settings}\label{libraries-and-settings}}
\addcontentsline{toc}{subsubsection}{Libraries and settings}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{options}\NormalTok{(}\StringTok{"repos"}\NormalTok{ =}\StringTok{ }\KeywordTok{list}\NormalTok{(}\DataTypeTok{CRAN=}\StringTok{"http://cran.rstudio.com/"}\NormalTok{))}

\KeywordTok{library}\NormalTok{(devtools)   }\CommentTok{# for revdp()}
\KeywordTok{library}\NormalTok{(igraph)     }\CommentTok{# for graph_from_edgelist/( and simplify() )}
\KeywordTok{library}\NormalTok{(visNetwork) }\CommentTok{# for visNetwork() and friends}
\KeywordTok{library}\NormalTok{(networkD3)  }\CommentTok{# for saveNetwork()}
\KeywordTok{library}\NormalTok{(chromote)   }\CommentTok{# for default_chromote_object()}
\KeywordTok{library}\NormalTok{(webshot2)   }\CommentTok{# for webshot()}
\KeywordTok{library}\NormalTok{(png)        }\CommentTok{# For displaying an image}
\KeywordTok{library}\NormalTok{(dplyr)}
\KeywordTok{library}\NormalTok{(purrr)}

\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/revDepNetHelper.R"}\NormalTok{)}

\KeywordTok{set_default_chromote_object}\NormalTok{(Chromote}\OperatorTok{$}\KeywordTok{new}\NormalTok{(}\DataTypeTok{browser =}\NormalTok{ Chrome}\OperatorTok{$}\KeywordTok{new}\NormalTok{(}\DataTypeTok{args =} \StringTok{"--no-sandbox"}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\hypertarget{read-package-names-from-our-table}{%
\subsubsection*{Read package names from our table}\label{read-package-names-from-our-table}}
\addcontentsline{toc}{subsubsection}{Read package names from our table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{reviewTables <-}\StringTok{ }\KeywordTok{read.delim}\NormalTok{(}\StringTok{"public/data/AllMetaRbolomicsTables.csv"}\NormalTok{, }\DataTypeTok{stringsAsFactors =} \OtherTok{FALSE}\NormalTok{)}
\NormalTok{reviewPkgs <-}\StringTok{ }\NormalTok{reviewTables[,}\StringTok{"Package"}\NormalTok{]}

\NormalTok{pkgs <-}\StringTok{ }\NormalTok{reviewPkgs}
\end{Highlighting}
\end{Shaded}

\hypertarget{get-reverse-dependencies}{%
\subsubsection*{Get reverse dependencies}\label{get-reverse-dependencies}}
\addcontentsline{toc}{subsubsection}{Get reverse dependencies}

\hypertarget{for-cran-and-bioc-packages}{%
\paragraph{For CRAN and BioC packages}\label{for-cran-and-bioc-packages}}
\addcontentsline{toc}{paragraph}{For CRAN and BioC packages}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{el <-}\StringTok{ }\KeywordTok{sapply}\NormalTok{(pkgs, }\ControlFlowTok{function}\NormalTok{(pkg) \{}
\NormalTok{  rd <-}\StringTok{ }\KeywordTok{revdep}\NormalTok{(pkg, }\DataTypeTok{dependencies =} \KeywordTok{c}\NormalTok{(}\StringTok{"Depends"}\NormalTok{, }\StringTok{"Imports"}\NormalTok{, }\StringTok{"LinkingTo"}\NormalTok{), }
               \DataTypeTok{recursive =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{ignore =} \OtherTok{NULL}\NormalTok{, }\DataTypeTok{bioconductor =} \OtherTok{TRUE}\NormalTok{)}
  \KeywordTok{as.matrix}\NormalTok{(}\KeywordTok{cbind}\NormalTok{(}\DataTypeTok{Package=}\KeywordTok{rep}\NormalTok{(pkg, }\DataTypeTok{length.out=}\KeywordTok{length}\NormalTok{(rd)), }\DataTypeTok{ReverseDep=}\NormalTok{rd))}
\NormalTok{\})}
\NormalTok{el <-}\StringTok{ }\KeywordTok{do.call}\NormalTok{(rbind, el)}
\end{Highlighting}
\end{Shaded}

\hypertarget{for-github-and-gitlab}{%
\paragraph{For GitHub and GitLab}\label{for-github-and-gitlab}}
\addcontentsline{toc}{paragraph}{For GitHub and GitLab}

The above \texttt{devtools::revdep} cannot read from GitHub/GitLab repositories.
We have a helper function that downloads and parses the DESCRIPTION file from GitHub/GitLab.
Since we cannot get reverse dependencies directly for GitHub/GitLab packages, those packages they are only used as additional reverse depedencies for the CRAN/BioC packages.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gitdeps_reverse <-}\StringTok{ }\NormalTok{reviewTables }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{dep_tree =} \KeywordTok{map}\NormalTok{(Code_link, get_git_deps)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{pull}\NormalTok{(dep_tree) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{bind_rows}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{filter}\NormalTok{(Dep }\OperatorTok{%in%}\StringTok{ }\NormalTok{el[,}\StringTok{"Package"}\NormalTok{]) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{rename}\NormalTok{(}\DataTypeTok{Package =}\NormalTok{ Dep, }\DataTypeTok{ReverseDep =}\NormalTok{ Package) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{as.matrix}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in readLines(file): incomplete final line found on '/tmp/
## RtmpKYNMBZ/file3bd639cc31cb'
\end{verbatim}

\begin{verbatim}
## Warning in readLines(file): incomplete final line found on '/tmp/
## RtmpKYNMBZ/file3bd6398fe9f5'
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{el <-}\StringTok{ }\KeywordTok{rbind}\NormalTok{(el, gitdeps_reverse)}
\end{Highlighting}
\end{Shaded}

\hypertarget{building-dependency-network}{%
\subsubsection*{Building dependency network}\label{building-dependency-network}}
\addcontentsline{toc}{subsubsection}{Building dependency network}

In total, we were analysing 292 packages.
For each package, this returns the set of packages in CRAN or BioC that depend on, import from or link to the package (i.e., its direct reverse dependencies) using the \texttt{devtools::revdep()} function. A few packages with the highest number of reverse dependencies have been excluded, as they would dominate the visualisation.
It was not possible to detect reverse dependencies from other hosting places such as GitHub or GitLab.

From the total, 68 packages had at least one
such reverse dependency.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{## Remove packages with most reverse dependencies }
\NormalTok{## which would dominate the network}

\NormalTok{el <-}\StringTok{ }\NormalTok{el[}\OperatorTok{!}\StringTok{ }\NormalTok{el[,}\StringTok{"Package"}\NormalTok{] }\OperatorTok{%in%}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"Rcpp"}\NormalTok{, }\StringTok{"igraph"}\NormalTok{, }\StringTok{"vegan"}\NormalTok{, }\StringTok{"caret"}\NormalTok{, }\StringTok{"rJava"}\NormalTok{, }\StringTok{"reticulate"}\NormalTok{), ]}

\NormalTok{## Create graph, and simplify redundancy}
\NormalTok{g <-}\StringTok{ }\KeywordTok{graph_from_edgelist}\NormalTok{(el, }\DataTypeTok{directed =} \OtherTok{TRUE}\NormalTok{)}
\NormalTok{g <-}\StringTok{ }\NormalTok{igraph}\OperatorTok{::}\KeywordTok{simplify}\NormalTok{(g, }\DataTypeTok{remove.multiple =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{remove.loops =} \OtherTok{TRUE}\NormalTok{)}

\CommentTok{# get data and plot :}
\NormalTok{data <-}\StringTok{ }\KeywordTok{toVisNetworkData}\NormalTok{(g)}

\NormalTok{data}\OperatorTok{$}\NormalTok{nodes <-}\StringTok{ }\KeywordTok{cbind}\NormalTok{(data}\OperatorTok{$}\NormalTok{nodes, }
                    \DataTypeTok{font.size=}\DecValTok{30}\NormalTok{, }
                    \DataTypeTok{color.background =} \KeywordTok{ifelse}\NormalTok{(data}\OperatorTok{$}\NormalTok{nodes[,}\StringTok{"id"}\NormalTok{] }\OperatorTok{%in%}\StringTok{ }\NormalTok{pkgs , }
                           \KeywordTok{rgb}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{0}\NormalTok{, }\DecValTok{200}\NormalTok{, }\DecValTok{128}\NormalTok{, }\DataTypeTok{max =} \DecValTok{255}\NormalTok{), }
                           \KeywordTok{rgb}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{200}\NormalTok{, }\DecValTok{0}\NormalTok{, }\DecValTok{128}\NormalTok{, }\DataTypeTok{max =} \DecValTok{255}\NormalTok{)))}

\NormalTok{vn <-}\StringTok{ }\KeywordTok{visNetwork}\NormalTok{(}\DataTypeTok{nodes =}\NormalTok{ data}\OperatorTok{$}\NormalTok{nodes, }
                 \DataTypeTok{edges =}\NormalTok{ data}\OperatorTok{$}\NormalTok{edges,}
                 \DataTypeTok{width=}\DecValTok{1000}\NormalTok{, }\DataTypeTok{height=}\DecValTok{1000}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{visPhysics}\NormalTok{(}\DataTypeTok{timestep =} \FloatTok{0.3}\NormalTok{,}
             \DataTypeTok{barnesHut =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{centralGravity=}\FloatTok{0.35}\NormalTok{,}
                              \DataTypeTok{springLength =} \DecValTok{95}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{visOptions}\NormalTok{(}\DataTypeTok{highlightNearest =} \OtherTok{TRUE}\NormalTok{)}

\NormalTok{vn}
\end{Highlighting}
\end{Shaded}

\includegraphics{The_MetaRbolomics_book_files/figure-latex/revVisNetwork-1.pdf}

Figure S1: Dependency network of R packages. Shown in blue are packages mentioned in the review. Edges connect to packages that depend on another package, as long as that is in CRAN or BioC. Green nodes correspond to packages in CRAN or BioC not covered in the review. Not shown are 1) infrastructure packages e.g.~rJava, Rcpp 2) packages from the review without reverse dependencies and 3) data packages. Some packages from the review are not in current versions of CRAN or BioC. An interactive version of this figure is available from \href{appendix-2-metarbolomics-dependencies-network.html}{https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html}.

\hypertarget{save-network-plot}{%
\subsubsection*{Save network plot}\label{save-network-plot}}
\addcontentsline{toc}{subsubsection}{Save network plot}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{saveNetwork}\NormalTok{(vn, }\StringTok{"vn.html"}\NormalTok{)}
\KeywordTok{webshot}\NormalTok{(}\StringTok{"vn.html"}\NormalTok{, }\StringTok{"revDepNet-60.png"}\NormalTok{, }\DataTypeTok{delay =} \DecValTok{60}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{The_MetaRbolomics_book_files/figure-latex/fileVis-1.png}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vnZoom <-}\StringTok{ }\KeywordTok{visNetwork}\NormalTok{(}\DataTypeTok{nodes =}\NormalTok{ data}\OperatorTok{$}\NormalTok{nodes, }
                 \DataTypeTok{edges =}\NormalTok{ data}\OperatorTok{$}\NormalTok{edges,}
                 \DataTypeTok{width=}\DecValTok{1000}\NormalTok{, }\DataTypeTok{height=}\DecValTok{1000}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{visIgraphLayout}\NormalTok{()}\OperatorTok{%>%}
\StringTok{    }\KeywordTok{visEvents}\NormalTok{(}\DataTypeTok{type=}\StringTok{"once"}\NormalTok{, }\DataTypeTok{startStabilizing =} \StringTok{'function() \{}
\StringTok{              this.fit(\{nodes:["ptw", "Rnmr1D", "RpeakChrom", "alsace", }
\StringTok{                               "PepsNMR", "ASICS", "MODIS", "RGCxGC"]\})}
\StringTok{            \}'}\NormalTok{)}
\KeywordTok{saveNetwork}\NormalTok{(vnZoom, }\StringTok{"vnZoom.html"}\NormalTok{)}
\KeywordTok{webshot}\NormalTok{(}\StringTok{"vnZoom.html"}\NormalTok{, }\StringTok{"revDepNet-zoom.png"}\NormalTok{, }\DataTypeTok{delay =} \DecValTok{5}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{The_MetaRbolomics_book_files/figure-latex/fileVisZoom-1.png}

You ca access the files at:

\begin{itemize}
\tightlist
\item
  \url{vn.html}
\item
  \url{revDepNet-60.png}
\item
  \url{vnZoom.html}
\item
  \url{revDepNet-zoom.png}
\end{itemize}

\hypertarget{notes}{%
\subsubsection*{Notes}\label{notes}}
\addcontentsline{toc}{subsubsection}{Notes}

The source code for this page is on GitHub at \href{https://gitlab.com/stanstrup/metaRbolomics-book}{gitlab.com/stanstrup/metaRbolomics-book}

The HTML output is shown at
\href{appendix-2-metarbolomics-dependencies-network.html}{https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html}

and \href{vn.html}{https://stanstrup.gitlab.io/metaRbolomics-book/vn.html} (Caveat: long rendering time, blank page without any visible progress)

This page was created with the following packages:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sessionInfo}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## R version 3.6.1 (2017-01-27)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.6 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] desc_1.2.0          png_0.1-7           webshot2_0.0.0.9000
##  [4] chromote_0.0.0.9001 networkD3_0.4       visNetwork_2.0.8   
##  [7] igraph_1.2.4.1      devtools_2.2.0      usethis_1.5.1      
## [10] tikzDevice_0.12.3   purrr_0.3.2         kableExtra_1.1.0   
## [13] DT_0.8              dplyr_0.8.3         googlesheets_0.3.0 
## [16] readr_1.3.1         knitr_1.24         
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.1          pkgload_1.0.2       jsonlite_1.6       
##  [4] viridisLite_0.3.0   shiny_1.3.2         assertthat_0.2.1   
##  [7] BiocManager_1.30.4  cellranger_1.1.0    yaml_2.2.0         
## [10] remotes_2.1.0       sessioninfo_1.1.1   pillar_1.4.2       
## [13] backports_1.1.4     glue_1.3.1          digest_0.6.20      
## [16] promises_1.0.1.9002 rvest_0.3.4         colorspace_1.4-1   
## [19] websocket_1.1.0     htmltools_0.3.6     httpuv_1.5.2       
## [22] pkgconfig_2.0.2     bookdown_0.13.2     xtable_1.8-4       
## [25] scales_1.0.0        webshot_0.5.1       processx_3.4.1     
## [28] later_0.8.0.9004    tibble_2.1.3        ellipsis_0.2.0.1   
## [31] withr_2.1.2         cli_1.1.0           magrittr_1.5       
## [34] crayon_1.3.4        mime_0.7            memoise_1.1.0      
## [37] evaluate_0.14       ps_1.3.0            fs_1.3.1           
## [40] xml2_1.2.2          pkgbuild_1.0.5      tools_3.6.1        
## [43] prettyunits_1.0.2   hms_0.5.1           stringr_1.4.0      
## [46] munsell_0.5.0       callr_3.3.1         compiler_3.6.1     
## [49] rlang_0.4.0         grid_3.6.1          rstudioapi_0.10    
## [52] htmlwidgets_1.3     filehash_2.4-2      crosstalk_1.0.0    
## [55] rmarkdown_1.15      testthat_2.2.1      codetools_0.2-16   
## [58] curl_4.0            R6_2.4.0            fastmap_1.0.0      
## [61] zeallot_0.1.0       rprojroot_1.3-2     stringi_1.4.3      
## [64] Rcpp_1.0.2          vctrs_0.2.0         tidyselect_0.2.5   
## [67] xfun_0.9
\end{verbatim}


\end{document}
