<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2.7 Network analysis and biochemical pathways | The MetaRbolomics book</title>
  <meta name="description" content="2.7 Network analysis and biochemical pathways | The MetaRbolomics book" />
  <meta name="generator" content="bookdown 0.13 and GitBook 2.6.7" />

  <meta property="og:title" content="2.7 Network analysis and biochemical pathways | The MetaRbolomics book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2.7 Network analysis and biochemical pathways | The MetaRbolomics book" />
  
  
  

<meta name="author" content="Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="handling-of-molecule-structures-and-chemical-structure-databases.html"/>
<link rel="next" href="multifunctional-workflows.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.8/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-rowgroup-1.10.16/css/rowGroup.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-rowgroup-1.10.16/js/dataTables.rowGroup.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="libs/vis-4.20.1/vis.min.js"></script>
<script src="libs/visNetwork-binding-2.0.8/visNetwork.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The MetaRbolomics book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="metabolomics-data-processing-and-analysis.html"><a href="metabolomics-data-processing-and-analysis.html"><i class="fa fa-check"></i><b>1.1</b> Metabolomics data processing and analysis</a></li>
<li class="chapter" data-level="1.2" data-path="the-r-package-landscape.html"><a href="the-r-package-landscape.html"><i class="fa fa-check"></i><b>1.2</b> The R package landscape</a></li>
<li class="chapter" data-level="1.3" data-path="dependences-and-connectivity-of-metabolomics-packages.html"><a href="dependences-and-connectivity-of-metabolomics-packages.html"><i class="fa fa-check"></i><b>1.3</b> Dependences and connectivity of metabolomics packages</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-packages-for-metabolomics.html"><a href="r-packages-for-metabolomics.html"><i class="fa fa-check"></i><b>2</b> R-packages for metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html"><i class="fa fa-check"></i><b>2.1</b> Mass spectrometry data handling and (pre-)processing</a><ul>
<li class="chapter" data-level="2.1.1" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#profile-mode-and-centroided-data"><i class="fa fa-check"></i><b>2.1.1</b> Profile mode and centroided data</a></li>
<li class="chapter" data-level="2.1.2" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#direct-infusion-mass-spectrometry-data"><i class="fa fa-check"></i><b>2.1.2</b> Direct infusion mass spectrometry data</a></li>
<li class="chapter" data-level="2.1.3" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#hyphenated-ms-and-non-targeted-data"><i class="fa fa-check"></i><b>2.1.3</b> Hyphenated MS and non-targeted data</a></li>
<li class="chapter" data-level="2.1.4" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#targeted-data-and-alternative-representations-of-data"><i class="fa fa-check"></i><b>2.1.4</b> Targeted data and alternative representations of data</a></li>
<li class="chapter" data-level="2.1.5" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#additional-dimensionality"><i class="fa fa-check"></i><b>2.1.5</b> Additional dimensionality</a></li>
<li class="chapter" data-level="2.1.6" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#structuring-data-and-metadata"><i class="fa fa-check"></i><b>2.1.6</b> Structuring data and metadata</a></li>
<li class="chapter" data-level="2.1.7" data-path="mass-spectrometry-data-handling-and-pre-processing.html"><a href="mass-spectrometry-data-handling-and-pre-processing.html#ion-species-grouping-and-annotation"><i class="fa fa-check"></i><b>2.1.7</b> Ion species grouping and annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="metabolite-identification-with-msms-data.html"><a href="metabolite-identification-with-msms-data.html"><i class="fa fa-check"></i><b>2.2</b> Metabolite identification with MS/MS data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="metabolite-identification-with-msms-data.html"><a href="metabolite-identification-with-msms-data.html#msms-data-handling-spectral-matching-and-clustering"><i class="fa fa-check"></i><b>2.2.1</b> MS/MS data handling, spectral matching and clustering</a></li>
<li class="chapter" data-level="2.2.2" data-path="metabolite-identification-with-msms-data.html"><a href="metabolite-identification-with-msms-data.html#reading-of-spectral-databases"><i class="fa fa-check"></i><b>2.2.2</b> Reading of spectral databases</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="nmr-data-handling-and-pre-processing.html"><a href="nmr-data-handling-and-pre-processing.html"><i class="fa fa-check"></i><b>2.3</b> NMR data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.4" data-path="uv-data-handling-and-pre-processing.html"><a href="uv-data-handling-and-pre-processing.html"><i class="fa fa-check"></i><b>2.4</b> UV data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.5" data-path="statistical-analysis-of-metabolomics-data.html"><a href="statistical-analysis-of-metabolomics-data.html"><i class="fa fa-check"></i><b>2.5</b> Statistical analysis of metabolomics data</a></li>
<li class="chapter" data-level="2.6" data-path="handling-of-molecule-structures-and-chemical-structure-databases.html"><a href="handling-of-molecule-structures-and-chemical-structure-databases.html"><i class="fa fa-check"></i><b>2.6</b> Handling of molecule structures and chemical structure databases</a></li>
<li class="chapter" data-level="2.7" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html"><i class="fa fa-check"></i><b>2.7</b> Network analysis and biochemical pathways</a><ul>
<li class="chapter" data-level="2.7.1" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html#network-infrastructure-and-analysis"><i class="fa fa-check"></i><b>2.7.1</b> Network infrastructure and analysis</a></li>
<li class="chapter" data-level="2.7.2" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html#metabolite-annotation"><i class="fa fa-check"></i><b>2.7.2</b> Metabolite annotation</a></li>
<li class="chapter" data-level="2.7.3" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html#generation-of-metabolic-networks"><i class="fa fa-check"></i><b>2.7.3</b> Generation of metabolic networks</a></li>
<li class="chapter" data-level="2.7.4" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html#pathway-analysis"><i class="fa fa-check"></i><b>2.7.4</b> Pathway analysis</a></li>
<li class="chapter" data-level="2.7.5" data-path="network-analysis-and-biochemical-pathways.html"><a href="network-analysis-and-biochemical-pathways.html#pathway-resources-and-interfaces"><i class="fa fa-check"></i><b>2.7.5</b> Pathway resources and interfaces</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="multifunctional-workflows.html"><a href="multifunctional-workflows.html"><i class="fa fa-check"></i><b>2.8</b> Multifunctional workflows</a></li>
<li class="chapter" data-level="2.9" data-path="user-interfaces-and-workflow-management-systems.html"><a href="user-interfaces-and-workflow-management-systems.html"><i class="fa fa-check"></i><b>2.9</b> User interfaces and workflow management systems</a></li>
<li class="chapter" data-level="2.10" data-path="metabolomics-data-sets.html"><a href="metabolomics-data-sets.html"><i class="fa fa-check"></i><b>2.10</b> Metabolomics data sets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>3</b> Conclusions</a></li>
<li class="chapter" data-level="4" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>4</b> Appendices</a><ul>
<li class="chapter" data-level="" data-path="appendix-1-the-msp-file-format-and-package-support.html"><a href="appendix-1-the-msp-file-format-and-package-support.html"><i class="fa fa-check"></i>Appendix 1: The MSP File Format and package support</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html"><i class="fa fa-check"></i>Appendix 2: metaRbolomics dependencies network</a><ul>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#libraries-and-settings"><i class="fa fa-check"></i>Libraries and settings</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#read-package-names-from-our-table"><i class="fa fa-check"></i>Read package names from our table</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#get-reverse-dependencies"><i class="fa fa-check"></i>Get reverse dependencies</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#building-dependency-network"><i class="fa fa-check"></i>Building dependency network</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#save-network-plot"><i class="fa fa-check"></i>Save network plot</a></li>
<li class="chapter" data-level="" data-path="appendix-2-metarbolomics-dependencies-network.html"><a href="appendix-2-metarbolomics-dependencies-network.html#notes"><i class="fa fa-check"></i>Notes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The MetaRbolomics book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="network-analysis-and-biochemical-pathways" class="section level2">
<h2><span class="header-section-number">2.7</span> Network analysis and biochemical pathways</h2>
<div id="network-infrastructure-and-analysis" class="section level3">
<h3><span class="header-section-number">2.7.1</span> Network infrastructure and analysis</h3>
</div>
<div id="metabolite-annotation" class="section level3">
<h3><span class="header-section-number">2.7.2</span> Metabolite annotation</h3>
</div>
<div id="generation-of-metabolic-networks" class="section level3">
<h3><span class="header-section-number">2.7.3</span> Generation of metabolic networks</h3>
</div>
<div id="pathway-analysis" class="section level3">
<h3><span class="header-section-number">2.7.4</span> Pathway analysis</h3>
</div>
<div id="pathway-resources-and-interfaces" class="section level3">
<h3><span class="header-section-number">2.7.5</span> Pathway resources and interfaces</h3>
<div id="htmlwidget-035789e04dd55ae6ae41" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-035789e04dd55ae6ae41">{"x":{"filter":"none","extensions":["RowGroup"],"caption":"<caption>Table 8: R packages for network analysis and Biochemical pathways.<\/caption>","fillContainer":false,"data":[["Network infrastructure and analysis","Network infrastructure and analysis","Network infrastructure and analysis","Network infrastructure and analysis","Network infrastructure and analysis","Network infrastructure and analysis","Annotation","Annotation","Annotation","Annotation","Annotation","Annotation","Generation of metabolite networks","Generation of metabolite networks","Generation of metabolite networks","Generation of metabolite networks","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway analysis","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces","Pathway resources and interfaces"],["Infrastructure for representation of networks, analysis and visualization.","Infrastructure for representation of networks, analysis and visualization.","Infrastructure for representation of networks, analysis and visualization.","Interactive visualization and manipulation of networks.","Comparison of correlation networks from two experiments.","Correlation-based networks from metabolomics data and analysis tools.","Putative annotation of unknowns in MS1 data.","Putative annotation of unknowns in MS1 data.","Putative annotation of unknowns using MS1 and MS2 data.","Visualization of spectral similarity networks, putative annotation of unknowns using MS2 data.","Putative annotation of unknowns using MS2 data, clustering of MS2 data.","Putative annotation of unknowns using MS2 data.","Biochemical reaction networks, spectral and structural similarity networks.","Correlation-based networks, structural similarity networks.","Targeted metabolome-wide association studies.","Generation of scale-free correlation-based networks.","Analysis of -omics data, pathway, transcription factor and target gene identification.","MSEA a metabolite set enrichment analysis with factor loading in principal component analysis.","Enrichment analysis of a list of affected metabolites.","Network-based enrichment analysis of a list of affected metabolites.","Pathway-based enrichment analysis of a list of affected metabolites.","Differential analysis, modules/sub-pathway identification using networks.","Integrates metabolic networks and RNA-seq data to construct condition-specific series of metabolic sub-networks and applies to gene set enrichment analysis","Differential analysis.","Biomarker identification.","Biomarker identification.","Biomarker identification.","Pathway activity profiling.","Pathway activity profiling.","Flux balance analysis.","Flux balance analysis.","Flux balance analysis.","Flux balance analysis.","Identification of affected pathway from phenotype data (interface with graphite).","Identification of affected pathway from phenotype data (interface with graphite).","Interface to PathVisio and WikiPathways and pathway analysis and enrichment.","Enrichment analysis of a list of genes and metabolites.","Simulation of longitudinal metabolomics data based on an underlying biological network","BioPax parser and representation in R.","Interface to KEGG, Biocarta, Reactome, NCI/Nature Pathway Interaction Database, HumanCyc, Panther, SMPDB and PharmGKB.","Interface to NCI Pathways Database.","Interface to KEGG.","Interface to KEGG.","Interface to systems biology markup language (SBML).","Interface to systems biology markup language (SBML).","Interface to Gaggle-enabled software (Cytoscape, Firegoose, Gaggle Genome browser).","Interface to molecular interaction databases.","Interface to KEGG REST server.","Interface to BioPAX OWL files and the Pathway Commons (PW) molecular interaction database.","Interface to WikiPathways.","Database that integrates metabolite and gene biological pathways from HMDB, KEGG, Reactome, and WikiPathways. Includes user-friendly R Shiny web application for queries and pathway enrichment analysis."],["<a target=_blank href=https://cran.r-project.org/package=igraph>igraph<\/a>","<a target=_blank href=https://cran.r-project.org/package=tidygraph>tidygraph<\/a>","<a target=_blank href=https://cran.r-project.org/package=statnet>statnet<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/RedeR.html>RedeR<\/a>","<a target=_blank href=https://cran.r-project.org/package=DiffCorr>DiffCorr<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/BioNetStat.html>BioNetStat<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/MetNet.html>MetNet<\/a>","<a target=_blank href=https://sourceforge.net/projects/xmsannotator/>xMSAnnotator<\/a>","<a target=_blank href=https://github.com/ZhuMSLab/MetDNA>MetDNA<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/MetCirc.html>MetCirc<\/a>","<a target=_blank href=https://bioconductor.org/packages/devel/bioc/html/CluMSID.html>CluMSID<\/a>","<a target=_blank href=https://github.com/WMBEdmands/compMS2Miner>compMS2Miner<\/a>","<a target=_blank href=http://dgrapov.github.io/MetaMapR/>MetaMapR<\/a>","<a target=_blank href=http://kwanjeeraw.github.io/metabox/>Metabox<\/a>","<a target=_blank href=https://sourceforge.net/projects/metabnet/>MetabNet<\/a>","<a target=_blank href=https://cran.r-project.org/package=WGCNA>WGCNA<\/a>","<a target=_blank href=NA>pwOmics<\/a>","<a target=_blank href=https://cran.r-project.org/package=mseapca>mseapca<\/a>","<a target=_blank href=https://cran.r-project.org/package=tmod>tmod<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/FELLA.html>FELLA<\/a>","<a target=_blank href=https://cran.r-project.org/package=CePa>CePa<\/a>","<a target=_blank href=https://github.com/andreasmock/MetaboDiff>MetaboDiff<\/a>","<a target=_blank href=https://cran.r-project.org/package=metaboGSE>metaboGSE<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/SDAMS.html>SDAMS<\/a>","<a target=_blank href=https://cran.r-project.org/package=lilikoi>lilikoi<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/INDEED.html>INDEED<\/a>","<a target=_blank href=https://github.com/krumsieklab/MoDentify>MoDentify<\/a>","<a target=_blank href=NA>PAPi<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/devel/bioc/html/pathwayPCA.html>pathwayPCA<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/BiGGR.html>BiGGR<\/a>","<a target=_blank href=https://cran.r-project.org/package=abcdeFBA>abcdeFBA<\/a>","<a target=_blank href=https://cran.r-project.org/package=sybil>sybil<\/a>","<a target=_blank href=https://cran.r-project.org/package=fbar>fbar<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/SPIA.html>SPIA<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/clipper.html>clipper<\/a>","<a target=_blank href=https://github.com/PathVisio/RpathVisio>RPathVisio<\/a>","<a target=_blank href=https://github.com/Mathelab/RaMP-DB>RaMP<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetaboLouise>MetaboLouise<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/rBiopaxParser.html>rBiopaxParser<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/graphite.html>graphite<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/NCIgraph.html>NCIgraph<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/pathview.html>pathview<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/KEGGgraph.html>KEGGgraph<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/SBMLR.html>SBMLR<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/rsbml.html>rsbml<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/gaggle.html>gaggle<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/PSICQUIC.html>PSICQUIC<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/KEGGREST.html>KEGGREST<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/paxtoolsr.html>paxtoolsr<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/rWikiPathways.html>rWikiPathways<\/a>","<a target=_blank href=https://github.com/Mathelab/RaMP-DB/>RaMP-DB<\/a>"],["CRAN","CRAN","CRAN","BioC","CRAN","BioC","BioC","SF","GitHub","BioC","BioC","GitHub","GitHub","GitHub","SF","CRAN","BioC","CRAN","CRAN","BioC","CRAN","GitHub","CRAN","BioC","CRAN","BioC","GitHub","BioC","BioC","BioC","CRAN","CRAN","CRAN","BioC","BioC","GitHub","GitHub","CRAN","BioC","BioC","BioC","BioC","BioC","BioC","BioC","BioC","BioC","BioC","BioC","BioC","GitHub"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Section<\/th>\n      <th>Functionalities<\/th>\n      <th>Package<\/th>\n      <th>Repo<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"rowGroup":{"dataSrc":0},"ordering":false,"columnDefs":[{"visible":false,"targets":0}],"pageLength":-1,"dom":"ft","order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[-1,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>

<div style="page-break-before: always;" />


</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="handling-of-molecule-structures-and-chemical-structure-databases.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multifunctional-workflows.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://gitlab.com/stanstrup/metaRbolomics-book/edit/master/0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["The_MetaRbolomics_book.pdf", "The_MetaRbolomics_book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
